<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single-cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single-cell RNA-seq data">
  <meta name="generator" content="bookdown 0.0.53 and GitBook 2.6.7">

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews and Martin Hemberg">

<meta name="date" content="2016-05-10">

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="introduction-to-differential-expression-de-analysis.html">
<link rel="next" href="de-expression-in-a-real-dataset.html">

<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.1</b> Registration</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.2</b> GitHub</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.4</b> Prerequisites</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> Spike-ins</a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a><ul>
<li class="chapter" data-level="4.5.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#umi-quantification"><i class="fa fa-check"></i><b>4.5.1</b> UMI quantification</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>5</b> scater package</a><ul>
<li class="chapter" data-level="5.1" data-path="scater-package.html"><a href="scater-package.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>5.2</b> scater workflow</a></li>
<li class="chapter" data-level="5.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>5.3</b> Terminology</a></li>
<li class="chapter" data-level="5.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>5.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>6</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>6.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>6.3</b> Cell QC</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>6.3.1</b> Library size</a></li>
<li class="chapter" data-level="6.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>6.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="6.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>6.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>6.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="6.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>6.4.1</b> Default Thresholds</a></li>
<li class="chapter" data-level="6.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>6.4.2</b> Automatic</a></li>
<li class="chapter" data-level="6.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>6.4.3</b> Manual</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>6.5</b> Compare filterings</a></li>
<li class="chapter" data-level="6.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>6.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="6.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>6.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="6.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>6.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>6.7</b> Save the data</a></li>
<li class="chapter" data-level="6.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>6.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>7</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="8" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>8</b> Data visualization</a><ul>
<li class="chapter" data-level="8.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>8.2</b> PCA plot</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>8.2.1</b> Before QC</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>8.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>8.3</b> tSNE map</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>8.3.1</b> Before QC</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>8.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>8.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-overview-reads.html"><a href="expression-overview-reads.html"><i class="fa fa-check"></i><b>9</b> Expression overview (Reads)</a></li>
<li class="chapter" data-level="10" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>10</b> Confounding factors</a><ul>
<li class="chapter" data-level="10.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>10.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="10.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>10.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>10.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="10.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>10.4</b> Other confounders</a></li>
<li class="chapter" data-level="10.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>10.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>11</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="12" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>12</b> Normalization for library size</a><ul>
<li class="chapter" data-level="12.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>12.2</b> Library size</a></li>
<li class="chapter" data-level="12.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>12.3</b> Normalisations</a><ul>
<li class="chapter" data-level="12.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>12.3.1</b> Raw</a></li>
<li class="chapter" data-level="12.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>12.3.2</b> CPM</a></li>
<li class="chapter" data-level="12.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>12.3.3</b> TMM</a></li>
<li class="chapter" data-level="12.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>12.3.4</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="12.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>12.3.5</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#other-methods"><i class="fa fa-check"></i><b>12.4</b> Other methods</a></li>
<li class="chapter" data-level="12.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>12.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="14" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html"><i class="fa fa-check"></i><b>14</b> Remove confounders using controls</a><ul>
<li class="chapter" data-level="14.1" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#introduction-5"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#brennecke-method"><i class="fa fa-check"></i><b>14.2</b> Brennecke method</a></li>
<li class="chapter" data-level="14.3" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>14.3</b> Remove Unwanted Variation</a></li>
<li class="chapter" data-level="14.4" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-1"><i class="fa fa-check"></i><b>14.4</b> Effectiveness 1</a></li>
<li class="chapter" data-level="14.5" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-2"><i class="fa fa-check"></i><b>14.5</b> Effectiveness 2</a></li>
<li class="chapter" data-level="14.6" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#exercise-2"><i class="fa fa-check"></i><b>14.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="remove-confounders-using-controls-reads.html"><a href="remove-confounders-using-controls-reads.html"><i class="fa fa-check"></i><b>15</b> Remove confounders using controls (Reads)</a></li>
<li class="chapter" data-level="16" data-path="clustering-analysis.html"><a href="clustering-analysis.html"><i class="fa fa-check"></i><b>16</b> Clustering analysis</a><ul>
<li class="chapter" data-level="16.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#introduction-6"><i class="fa fa-check"></i><b>16.1</b> Introduction</a></li>
<li class="chapter" data-level="16.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#dimensionality-reductions"><i class="fa fa-check"></i><b>16.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="16.2.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pca"><i class="fa fa-check"></i><b>16.2.1</b> PCA</a></li>
<li class="chapter" data-level="16.2.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#spectral"><i class="fa fa-check"></i><b>16.2.2</b> Spectral</a></li>
<li class="chapter" data-level="16.2.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne"><i class="fa fa-check"></i><b>16.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#clustering-methods"><i class="fa fa-check"></i><b>16.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="16.3.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>16.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="16.3.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#k-means"><i class="fa fa-check"></i><b>16.3.2</b> k-means</a></li>
<li class="chapter" data-level="16.3.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#graph-based-methods"><i class="fa fa-check"></i><b>16.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#challenges-in-clustering"><i class="fa fa-check"></i><b>16.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="16.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>16.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="16.5.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sincera"><i class="fa fa-check"></i><b>16.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="16.5.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pcareduce"><i class="fa fa-check"></i><b>16.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="16.5.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sc3"><i class="fa fa-check"></i><b>16.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="16.5.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne-k-means"><i class="fa fa-check"></i><b>16.5.4</b> tSNE + <em>k</em>-means</a></li>
<li class="chapter" data-level="16.5.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#snn-cliq"><i class="fa fa-check"></i><b>16.5.5</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>17</b> Clustering analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>17.1</b> Patient dataset</a></li>
<li class="chapter" data-level="17.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>17.2</b> SC3</a></li>
<li class="chapter" data-level="17.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>17.3</b> pcaReduce</a></li>
<li class="chapter" data-level="17.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>17.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="17.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>17.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="17.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>17.6</b> SINCERA</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>18</b> Introduction to differential expression (DE) analysis</a><ul>
<li class="chapter" data-level="18.1" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#background-bulk-rna-seq"><i class="fa fa-check"></i><b>18.1</b> Background (bulk RNA-seq)</a></li>
<li class="chapter" data-level="18.2" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#de-analysis-for-scrna-seq"><i class="fa fa-check"></i><b>18.2</b> DE analysis for scRNA-seq</a></li>
<li class="chapter" data-level="18.3" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#synthetic-data-for-scrna-seq"><i class="fa fa-check"></i><b>18.3</b> Synthetic data for scRNA-seq</a></li>
<li class="chapter" data-level="18.4" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#modelled-expression-of-a-single-gene"><i class="fa fa-check"></i><b>18.4</b> Modelled expression of a single gene</a></li>
<li class="chapter" data-level="18.5" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#possible-shapes-of-the-poisson-beta-distribution"><i class="fa fa-check"></i><b>18.5</b> Possible shapes of the Poisson-Beta distribution</a></li>
<li class="chapter" data-level="18.6" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#effect-of-sampling-size-on-parameter-estimates"><i class="fa fa-check"></i><b>18.6</b> Effect of sampling size on parameter estimates</a></li>
<li class="chapter" data-level="18.7" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#drop-out-noise"><i class="fa fa-check"></i><b>18.7</b> Drop-out noise</a></li>
<li class="chapter" data-level="18.8" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#over--and-under-dispersion-noise"><i class="fa fa-check"></i><b>18.8</b> Over- and under-dispersion noise</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html"><i class="fa fa-check"></i><b>19</b> DE expression in a synthetic dataset</a><ul>
<li class="chapter" data-level="19.1" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#generation-of-a-synthetic-dataset"><i class="fa fa-check"></i><b>19.1</b> Generation of a synthetic dataset</a></li>
<li class="chapter" data-level="19.2" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#differential-expression-in-scrna-seq"><i class="fa fa-check"></i><b>19.2</b> Differential expression in scRNA-seq</a></li>
<li class="chapter" data-level="19.3" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>19.3</b> Kolmogorov-Smirnov test</a></li>
<li class="chapter" data-level="19.4" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#performance-of-the-ks-test"><i class="fa fa-check"></i><b>19.4</b> Performance of the KS test</a></li>
<li class="chapter" data-level="19.5" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#finding-de-genes-using-deseq2"><i class="fa fa-check"></i><b>19.5</b> Finding DE genes using DESeq2</a></li>
<li class="chapter" data-level="19.6" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#finding-de-genes-using-scde"><i class="fa fa-check"></i><b>19.6</b> Finding DE genes using SCDE</a></li>
<li class="chapter" data-level="19.7" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>19.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="19.8" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#beyond-changes-in-the-mean"><i class="fa fa-check"></i><b>19.8</b> Beyond changes in the mean</a></li>
<li class="chapter" data-level="19.9" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#further-comparisons"><i class="fa fa-check"></i><b>19.9</b> Further comparisons</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html"><i class="fa fa-check"></i><b>20</b> DE expression in a real dataset</a><ul>
<li class="chapter" data-level="20.1" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#introduction-7"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#ks-test"><i class="fa fa-check"></i><b>20.2</b> KS-test</a></li>
<li class="chapter" data-level="20.3" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#deseq2"><i class="fa fa-check"></i><b>20.3</b> DESeq2</a></li>
<li class="chapter" data-level="20.4" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#scde"><i class="fa fa-check"></i><b>20.4</b> SCDE</a></li>
<li class="chapter" data-level="20.5" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#comparison-of-the-methods-1"><i class="fa fa-check"></i><b>20.5</b> Comparison of the methods</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>21</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single-cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="de-expression-in-a-synthetic-dataset" class="section level1">
<h1><span class="header-section-number">19</span> DE expression in a synthetic dataset</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scRNA.seq.funcs)
<span class="kw">library</span>(DESeq2)
<span class="kw">library</span>(scde)
<span class="kw">library</span>(ROCR)
<span class="kw">library</span>(limma)
<span class="kw">set.seed</span>(<span class="dv">1</span>)</code></pre></div>
<div id="generation-of-a-synthetic-dataset" class="section level2">
<h2><span class="header-section-number">19.1</span> Generation of a synthetic dataset</h2>
<p>We start by generating samples using the Poisson-Beta distribution, using 100 genes from 50 cells. To simulate the second experimental condition we select one of the three parameters for each gene and modify it by multiplying by a normally distributed random factor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nGenes &lt;-<span class="st"> </span><span class="fl">1e2</span>
nCells &lt;-<span class="st"> </span><span class="dv">50</span>
mult &lt;-<span class="st"> </span><span class="dv">2</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">0</span>, <span class="dv">2</span>))
synData &lt;-<span class="st"> </span>scRNA.seq.funcs::<span class="kw">GeneratePoiBetaSamples</span>(
    <span class="dt">ks =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">3</span>, .<span class="dv">5</span>)),
    <span class="dt">as =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, -<span class="dv">1</span>, .<span class="dv">5</span>)),
    <span class="dt">bs =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">0</span>, .<span class="dv">5</span>)),
    mult,
    nGenes,
    nCells
)
g &lt;-<span class="st"> </span>synData$sample1
g2 &lt;-<span class="st"> </span>synData$sample2</code></pre></div>
</div>
<div id="differential-expression-in-scrna-seq" class="section level2">
<h2><span class="header-section-number">19.2</span> Differential expression in scRNA-seq</h2>
<p>For bulk data, each gene is represented by a single value and to identify DEGs we need to identify those genes where the difference in expression between two conditions is sufficiently large. Replicates are needed for us to be able to assess the fold-change as well as its significance.</p>
<p>For single-cell data, the situation is more complicated; instead of comparing two means we are faced with the task of comparing two probability distributions. There are many different functions available for comparing two probability distributions (e.g. <a href="https://en.wikipedia.org/wiki/Total_variation_distance_of_probability_measures">Total variation distance</a> and <a href="https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence">Kullback-Leibler divergence</a>), and since they emphasize different features, they have different properties.</p>
<p>To establish a ground-truth, we arbitrarily assign genes where one of the parameters has changed by more than a factor of 4 as being true positives. Remaining genes are considered not significantly changed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">changedGenes &lt;-<span class="st"> </span><span class="kw">abs</span>(<span class="kw">log2</span>(mult)) &gt;<span class="st"> </span><span class="dv">2</span>
changedGenesInds &lt;-<span class="st"> </span><span class="kw">which</span>(changedGenes)
notChangedGenesInds &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">abs</span>(<span class="kw">log2</span>(mult)) &lt;=<span class="st"> </span><span class="dv">2</span>)</code></pre></div>
</div>
<div id="kolmogorov-smirnov-test" class="section level2">
<h2><span class="header-section-number">19.3</span> Kolmogorov-Smirnov test</h2>
<p>The types of test that are easiest to work with are non-parametric ones. The most commonly used non-parametric test is the <a href="https://en.wikipedia.org/wiki/Kolmogorov%E2%80%93Smirnov_test">Kolmogorov-Smirnov test</a> (KS-test) and we can use it to compare the distributions for each gene in the two conditions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nGenes &lt;-<span class="st"> </span><span class="kw">nrow</span>(g)
pVals &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="dv">1</span>, nGenes)
for (i in <span class="dv">1</span>:nGenes) {
    res &lt;-<span class="st"> </span><span class="kw">ks.test</span>(g[i,], g2[i,])
    <span class="co"># Bonferroni correction</span>
    pVals[i] &lt;-<span class="st"> </span>res$p.value*nGenes
}</code></pre></div>
<p>Using the standard p-value cut-off .05, we can find out how many genes that were called as significantly different.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ksChangedGenes &lt;-<span class="st"> </span><span class="kw">which</span>(pVals &lt;<span class="st"> </span><span class="fl">0.05</span>)
ksNotChangedGenes &lt;-<span class="st"> </span><span class="kw">which</span>(pVals &gt;=<span class="st"> </span><span class="fl">0.05</span>)
<span class="kw">cat</span>(changedGenesInds)</code></pre></div>
<pre><code>## 4 11 14 15 24 28 31 35 39 54 55 56 58 61 67 68 70 75 83 84 87 92 93 95 97 99</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cat</span>(ksChangedGenes)</code></pre></div>
<pre><code>## 13 28 31 35 40 55 56 61 67 84 87 97</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cat</span>(<span class="kw">intersect</span>(changedGenesInds, ksChangedGenes))</code></pre></div>
<pre><code>## 28 31 35 55 56 61 67 84 87 97</code></pre>
</div>
<div id="performance-of-the-ks-test" class="section level2">
<h2><span class="header-section-number">19.4</span> Performance of the KS test</h2>
<p>The genes identified by the KS-test differs substantially from the ground truth. Instead of considering the absolute number of identified genes, it is often more informative to consider <a href="https://en.wikipedia.org/wiki/False_positive_rate">False positive rate</a> (FPR) and the <a href="https://en.wikipedia.org/wiki/Sensitivity_and_specificity">True positive rate</a> (TPR). The False positive rate is defined as FPR = FP/(FP + TP) and the True positive rate as TPR = TP/(TP + FN), where FP is the number of false positives, TN the number of true negatives, TP the number of true negatives and FN the number of false negatives.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(changedGenesInds, ksChangedGenes))
fn &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(changedGenesInds, <span class="kw">which</span>(pVals &gt;=<span class="st"> </span>.<span class="dv">05</span>)))
fp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(notChangedGenesInds, ksChangedGenes))
tn &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(notChangedGenesInds, <span class="kw">which</span>(pVals &gt;=<span class="st"> </span>.<span class="dv">05</span>)))
tpr &lt;-<span class="st"> </span>tp/(tp +<span class="st"> </span>fn)
fpr &lt;-<span class="st"> </span>fp/(fp +<span class="st"> </span>tn)
<span class="kw">cat</span>(<span class="kw">c</span>(tpr, fpr))</code></pre></div>
<pre><code>## 0.3846154 0.02702703</code></pre>
<p>As you can see, the p-value cut-off .05 results in a low TPR and a high FPR. That is, the test has failed to identify many of the genes that were truly changed and many of the changed genes were not detected.</p>
<p>Clearly, there is a trade-off between TPR and FPR. If one is willing to accept a higher FPR, then one will be able to achieve a higher TPR. The relationship between FPR and TPR is typically shown as a receiver-operator-characteristic (ROC) curve. To generate and plot the ROC curve, we need to change the p-value cut-off.</p>
<p>To facilitate the plotting, we use the package “ROCR”</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pred &lt;-<span class="st"> </span>ROCR::<span class="kw">prediction</span>(pVals, <span class="kw">as.numeric</span>(<span class="kw">abs</span>(<span class="kw">log2</span>(mult)) &lt;=<span class="st"> </span><span class="dv">2</span>))
perf &lt;-<span class="st"> </span>ROCR::<span class="kw">performance</span>(pred, <span class="st">&quot;tpr&quot;</span>, <span class="st">&quot;fpr&quot;</span>)
ROCR::<span class="kw">plot</span>(perf)  </code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:synthetic-roc-curve-ks"></span>
<img src="18-de-synth_files/figure-html/synthetic-roc-curve-ks-1.png" alt="Different distributions of read counts for a single genes across 50 cells based on the Poisson-Beta model corresponding to different paramete sets"  />
<p class="caption">
Figure 19.1: Different distributions of read counts for a single genes across 50 cells based on the Poisson-Beta model corresponding to different paramete sets
</p>
</div>
<p>Often we are interested in comparing several ROC curves. To carry out such a comparison, we need to summarize the entire curve using only one scalar value. This can be achieved by calculating the area under the ROC curve (AUROC). Since an ROC curve has to stay above the diagonal (why?) the AUROC will be between .5 and 1.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">aucObj &lt;-<span class="st"> </span>ROCR::<span class="kw">performance</span>(pred, <span class="st">&quot;auc&quot;</span>)
aucObj@y.values[[<span class="dv">1</span>]]</code></pre></div>
<pre><code>## [1] 0.7837838</code></pre>
<p><strong>Exercise</strong>: Compare the AUC values when you change some of the parameters in the analysis, e.g. number of cells, number of genes, threshold for considering a gene differentially expressed, distribution of parameter values, distribution of fold-changes. What factors make it easy or hard to identify differentially expressed genes?</p>
</div>
<div id="finding-de-genes-using-deseq2" class="section level2">
<h2><span class="header-section-number">19.5</span> Finding DE genes using DESeq2</h2>
<p>One could still apply bulk DE methods to scRNA-seq data. One of the most popular methods for differential expression analysis for bulk RNA-Seq data is <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>. Let’s try it out on our synthetic dataset:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cnts &lt;-<span class="st"> </span><span class="kw">cbind</span>(g, g2)
cond &lt;-<span class="st"> </span><span class="kw">factor</span>(
    <span class="kw">c</span>(
        <span class="kw">rep</span>(<span class="st">&quot;A&quot;</span>, <span class="kw">ncol</span>(g)),
        <span class="kw">rep</span>(<span class="st">&quot;B&quot;</span>, <span class="kw">ncol</span>(g2))
    )
)
<span class="co"># object construction, add a pseudo-count of 1 to make DESeq work</span>
dds &lt;-<span class="st"> </span>DESeq2::<span class="kw">DESeqDataSetFromMatrix</span>(
    cnts +<span class="st"> </span><span class="dv">1</span>, 
    <span class="kw">DataFrame</span>(cond), 
    ~<span class="st"> </span>cond)
dds &lt;-<span class="st"> </span>DESeq2::<span class="kw">DESeq</span>(dds)
resDESeq &lt;-<span class="st"> </span><span class="kw">results</span>(dds)</code></pre></div>
<p>Check the performance of DESeq2:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pValsDESeq &lt;-<span class="st"> </span>resDESeq$padj
predDESeq &lt;-<span class="st"> </span><span class="kw">prediction</span>(pValsDESeq, <span class="kw">as.numeric</span>(<span class="kw">abs</span>(<span class="kw">log2</span>(mult)) &lt;=<span class="st"> </span><span class="dv">2</span>))
perfDESeq &lt;-<span class="st"> </span><span class="kw">performance</span>(predDESeq, <span class="st">&quot;tpr&quot;</span>, <span class="st">&quot;fpr&quot;</span>)
ROCR::<span class="kw">plot</span>(perfDESeq)  </code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:synthetic-roc-curve-deseq"></span>
<img src="18-de-synth_files/figure-html/synthetic-roc-curve-deseq-1.png" alt="Different distributions of read counts for a single genes across 50 cells based on the Poisson-Beta model corresponding to different paramete sets"  />
<p class="caption">
Figure 19.2: Different distributions of read counts for a single genes across 50 cells based on the Poisson-Beta model corresponding to different paramete sets
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">aucObjDESeq &lt;-<span class="st"> </span><span class="kw">performance</span>(predDESeq, <span class="st">&quot;auc&quot;</span>)
aucObjDESeq@y.values[[<span class="dv">1</span>]]</code></pre></div>
<pre><code>## [1] 0.8503119</code></pre>
<p><strong>Exercise:</strong> Based on the AUC-value, does DESeq or the KS-test seem more accurate? Can you find a parameter regime where the ranking is changed?</p>
</div>
<div id="finding-de-genes-using-scde" class="section level2">
<h2><span class="header-section-number">19.6</span> Finding DE genes using SCDE</h2>
<p>There are yet much fewer methods available for scRNA-seq data than for bulk data, but one popular method is <a href="http://hms-dbmi.github.io/scde/index.html">SCDE</a>. We can use it on the synthetic data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cnts &lt;-<span class="st"> </span><span class="kw">cbind</span>(g, g2)
cnts &lt;-<span class="st"> </span><span class="kw">apply</span>(
    cnts,
    <span class="dv">2</span>, 
    function(x) {
        <span class="kw">storage.mode</span>(x) &lt;-<span class="st"> &#39;integer&#39;</span>
        <span class="kw">return</span>(x)
    }
)
cond &lt;-<span class="st"> </span><span class="kw">factor</span>(
    <span class="kw">c</span>(
        <span class="kw">rep</span>(<span class="st">&quot;A&quot;</span>, <span class="kw">ncol</span>(g)),
        <span class="kw">rep</span>(<span class="st">&quot;B&quot;</span>, <span class="kw">ncol</span>(g2))
    )
)
<span class="kw">names</span>(cond) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">length</span>(cnts[<span class="dv">1</span>, ])
<span class="kw">colnames</span>(cnts) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">length</span>(cnts[<span class="dv">1</span>, ]) 
o.ifm &lt;-<span class="st"> </span>scde::<span class="kw">scde.error.models</span>(
    <span class="dt">counts =</span> cnts,
    <span class="dt">groups =</span> cond,
    <span class="dt">n.cores =</span> <span class="dv">1</span>,
    <span class="dt">threshold.segmentation =</span> <span class="ot">TRUE</span>,
    <span class="dt">save.crossfit.plots =</span> <span class="ot">FALSE</span>,
    <span class="dt">save.model.plots =</span> <span class="ot">FALSE</span>,
    <span class="dt">verbose =</span> <span class="dv">0</span>,
    <span class="dt">min.size.entries =</span> <span class="dv">20</span>
)
priors &lt;-<span class="st"> </span>scde::<span class="kw">scde.expression.prior</span>(
    <span class="dt">models =</span> o.ifm,
    <span class="dt">counts =</span> cnts,
    <span class="dt">length.out =</span> <span class="dv">400</span>,
    <span class="dt">show.plot =</span> <span class="ot">FALSE</span>)
resSCDE1 &lt;-<span class="st"> </span>scde::<span class="kw">scde.expression.difference</span>(
    o.ifm,
    cnts,
    priors,
    <span class="dt">groups =</span> cond,
    <span class="dt">n.randomizations =</span> <span class="dv">100</span>,
    <span class="dt">n.cores =</span> <span class="dv">1</span>,
    <span class="dt">verbose =</span> <span class="dv">0</span>)
pValsSCDE1 &lt;-<span class="st"> </span><span class="kw">pnorm</span>(resSCDE1$cZ, <span class="dt">lower.tail =</span> <span class="ot">FALSE</span>) 
<span class="co"># Need to run the other way as well </span>
cnts2 &lt;-<span class="st"> </span><span class="kw">cbind</span>(g2, g)
cnts2 &lt;-<span class="st"> </span><span class="kw">apply</span>(
    cnts2, 
    <span class="dv">2</span>, 
    function(x) {
        <span class="kw">storage.mode</span>(x) &lt;-<span class="st"> &#39;integer&#39;</span>
        <span class="kw">return</span>(x)
    }
)
<span class="kw">names</span>(cond) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">length</span>(cnts2[<span class="dv">1</span>, ])
<span class="kw">colnames</span>(cnts2) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">length</span>(cnts2[<span class="dv">1</span>, ]) 
o.ifm &lt;-<span class="st"> </span>scde::<span class="kw">scde.error.models</span>(
    <span class="dt">counts =</span> cnts2,
    <span class="dt">groups =</span> cond,
    <span class="dt">n.cores =</span> <span class="dv">1</span>,
    <span class="dt">threshold.segmentation =</span> <span class="ot">TRUE</span>,
    <span class="dt">save.crossfit.plots =</span> <span class="ot">FALSE</span>,
    <span class="dt">save.model.plots =</span> <span class="ot">FALSE</span>,
    <span class="dt">verbose =</span> <span class="dv">0</span>,
    <span class="dt">min.size.entries =</span> <span class="dv">20</span>
)
priors &lt;-<span class="st"> </span>scde::<span class="kw">scde.expression.prior</span>(
    <span class="dt">models =</span> o.ifm,
    <span class="dt">counts =</span> cnts2,
    <span class="dt">length.out =</span> <span class="dv">400</span>,
    <span class="dt">show.plot =</span> <span class="ot">FALSE</span>
)
resSCDE2 &lt;-<span class="st"> </span>scde::<span class="kw">scde.expression.difference</span>(
    o.ifm,
    cnts2,
    priors,
    <span class="dt">groups =</span> cond,
    <span class="dt">n.randomizations =</span> <span class="dv">100</span>,
    <span class="dt">n.cores =</span> <span class="dv">1</span>,
    <span class="dt">verbose =</span> <span class="dv">0</span>
)
pValsSCDE2 &lt;-<span class="st"> </span><span class="kw">pnorm</span>(resSCDE2$cZ, <span class="dt">lower.tail =</span> <span class="ot">FALSE</span>) </code></pre></div>
<p>As you can see, SCDE is different from the KS-test and DESeq2 since it only searches for changes in one direction. Thus, if we want to find both upregulated and downregulated genes, we must run the software twice and then combine the results as we have done above.</p>
<p><strong>Exercise:</strong> Merge the results from the two runs and produce a single list of p-values and (absolute) fold-changes for all genes. Based on these results, calculate an AUROC value and compare to the other methods.</p>
<p><strong>Hint:</strong> Keep in mind that we are now doing two tests for each gene.</p>
<p><strong>Our answer</strong>: <img src="18-de-synth_files/figure-html/unnamed-chunk-8-1.png" title="" alt="" style="display: block; margin: auto;" /></p>
<pre><code>## [1] 0.8607069</code></pre>
</div>
<div id="comparison-of-the-methods" class="section level2">
<h2><span class="header-section-number">19.7</span> Comparison of the methods</h2>
<p>From the above analyses it is clear that none of the three methods (KS, DESeq2 and SCDE) is able to reliably find the majority of the DE genes. A popular strategy in this situation is to combine two or more methods, in the hope that genes that were identified by more than method are more likely to be true positives.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))
limma::<span class="kw">vennDiagram</span>(
    <span class="kw">vennCounts</span>(
        <span class="kw">cbind</span>(
            <span class="dv">1</span> -<span class="st"> </span>changedGenes,
            pValsDESeq &lt;<span class="st"> </span>.<span class="dv">05</span>,
            pValsSCDE &lt;<span class="st"> </span>.<span class="dv">05</span>,
            pVals &lt;<span class="st"> </span>.<span class="dv">05</span>
        )
    ),
    <span class="dt">names =</span> <span class="kw">c</span>(<span class="st">&quot;Ground truth&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;SCDE&quot;</span>, <span class="st">&quot;KS-test&quot;</span>),
    <span class="dt">circle.col =</span> <span class="kw">c</span>(<span class="st">&quot;magenta&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</code></pre></div>
<p><img src="18-de-synth_files/figure-html/synthetic-differentially-expressed-combination-1.png" title="" alt="" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">allChangedInds &lt;-<span class="st"> </span><span class="kw">intersect</span>(
    <span class="kw">which</span>(pValsDESeq &lt;<span class="st"> </span>.<span class="dv">05</span>),
    <span class="kw">intersect</span>(<span class="kw">which</span>(pValsSCDE &lt;<span class="st"> </span>.<span class="dv">05</span>),
              <span class="kw">which</span>(pVals &lt;<span class="st"> </span>.<span class="dv">05</span>)
    )
)
tpAll &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(changedGenesInds, allChangedInds))
fnAll &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(changedGenesInds, <span class="kw">setdiff</span>(<span class="dv">1</span>:<span class="fl">1e3</span>, allChangedInds)))
fpAll &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(notChangedGenesInds, allChangedInds))
tnAll &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(notChangedGenesInds, <span class="kw">setdiff</span>(<span class="dv">1</span>:<span class="fl">1e3</span>, allChangedInds)))
tprAll &lt;-<span class="st"> </span>tpAll /<span class="st"> </span>(tpAll +<span class="st"> </span>fnAll)
fprAll &lt;-<span class="st"> </span>fpAll /<span class="st"> </span>(fpAll +<span class="st"> </span>tnAll)
<span class="kw">cat</span>(<span class="kw">c</span>(tprAll, fprAll))</code></pre></div>
<pre><code>## 0.3846154 0.01351351</code></pre>
<p>The more stringent approach results in a lower FPR at the cost of a lower TPR.</p>
<p><strong>Exercise:</strong> Calculate the TPR and FPR for the case when we require two out of three methods to agree.</p>
</div>
<div id="beyond-changes-in-the-mean" class="section level2">
<h2><span class="header-section-number">19.8</span> Beyond changes in the mean</h2>
<p>In the realm of single-cell analyses, differential expression is no longer restricted to changes in the mean. As we saw in the previous chapter, it is possible to change the shape of the distribution without changing its mean.</p>
<p><strong>Exercise:</strong> Repeat the analysis above, but this time only change the variance, but not the mean. To generate the synthetic data, use the following command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nGenes &lt;-<span class="st"> </span><span class="fl">1e2</span>
nCells &lt;-<span class="st"> </span><span class="dv">50</span>
mult &lt;-<span class="st"> </span><span class="dv">2</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">0</span>, <span class="dv">2</span>))
synData &lt;-<span class="st"> </span>scRNA.seq.funcs::<span class="kw">GeneratePoiBetaSamples</span>(
    <span class="dt">ks =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">3</span>, .<span class="dv">5</span>)),
    <span class="dt">as =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, -<span class="dv">1</span>, .<span class="dv">5</span>)),
    <span class="dt">bs =</span> <span class="dv">10</span>^(<span class="kw">rnorm</span>(nGenes, <span class="dv">0</span>, .<span class="dv">5</span>)),
    mult,
    nGenes,
    nCells,
    <span class="dt">meanFixed =</span> <span class="ot">TRUE</span>
)
g &lt;-<span class="st"> </span>synData$sample1
g2 &lt;-<span class="st"> </span>synData$sample2</code></pre></div>
<p><strong>Our answer</strong>: <img src="18-de-synth_files/figure-html/unnamed-chunk-9-1.png" title="" alt="" style="display: block; margin: auto;" /></p>
<pre><code>## [1] 0.8123021</code></pre>
<p><img src="18-de-synth_files/figure-html/unnamed-chunk-9-2.png" title="" alt="" style="display: block; margin: auto;" /></p>
<pre><code>## [1] 0.475142</code></pre>
<p><img src="18-de-synth_files/figure-html/unnamed-chunk-9-3.png" title="" alt="" style="display: block; margin: auto;" /></p>
<pre><code>## [1] 0.4312528</code></pre>
<p>Which method performs the best? Why do some methods perform differently for this scenario? What are the properties of the genes that are identified as DE?</p>
</div>
<div id="further-comparisons" class="section level2">
<h2><span class="header-section-number">19.9</span> Further comparisons</h2>
<p>To get more confidence in saying which method is better, the methods can be compared multiple times.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-differential-expression-de-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="de-expression-in-a-real-dataset.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/18-de-synth.Rmd",
"text": "Edit"
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>


</body>

</html>
