<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single-cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single-cell RNA-seq data">
  <meta name="generator" content="bookdown 0.0.53 and GitBook 2.6.7">

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews and Martin Hemberg">

<meta name="date" content="2016-03-23">

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="scater-package.html">
<link rel="next" href="expression-qc-reads.html">

<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.1</b> Registration</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.2</b> GitHub</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.4</b> Prerequisites</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#analysis"><i class="fa fa-check"></i><b>3.4</b> Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#ercc-spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> ERCC <em>spike-ins</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>5</b> scater package</a><ul>
<li class="chapter" data-level="5.1" data-path="scater-package.html"><a href="scater-package.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>5.2</b> scater workflow</a></li>
<li class="chapter" data-level="5.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>5.3</b> Terminology</a></li>
<li class="chapter" data-level="5.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>5.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>6</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>6.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-qc"><i class="fa fa-check"></i><b>6.3</b> Gene QC</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-dropouts"><i class="fa fa-check"></i><b>6.3.1</b> Gene dropouts</a></li>
<li class="chapter" data-level="6.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>6.3.2</b> Gene expression</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>6.4</b> Gene filtering</a></li>
<li class="chapter" data-level="6.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>6.5</b> Cell QC</a><ul>
<li class="chapter" data-level="6.5.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>6.5.1</b> Library size</a></li>
<li class="chapter" data-level="6.5.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>6.5.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="6.5.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-2"><i class="fa fa-check"></i><b>6.5.3</b> Detected genes (2)</a></li>
<li class="chapter" data-level="6.5.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>6.5.4</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>6.6</b> Cell filtering</a><ul>
<li class="chapter" data-level="6.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default"><i class="fa fa-check"></i><b>6.6.1</b> Default</a></li>
<li class="chapter" data-level="6.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>6.6.2</b> Automatic</a></li>
<li class="chapter" data-level="6.6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>6.6.3</b> Manual</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>6.7</b> Compare filterings</a></li>
<li class="chapter" data-level="6.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>6.8</b> Save the data</a></li>
<li class="chapter" data-level="6.9" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#exercise"><i class="fa fa-check"></i><b>6.9</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>7</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="8" data-path="expression-overview.html"><a href="expression-overview.html"><i class="fa fa-check"></i><b>8</b> Expression overview</a><ul>
<li class="chapter" data-level="8.1" data-path="expression-overview.html"><a href="expression-overview.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="expression-overview.html"><a href="expression-overview.html#top-500-genes"><i class="fa fa-check"></i><b>8.2</b> Top 500 genes</a><ul>
<li class="chapter" data-level="8.2.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc"><i class="fa fa-check"></i><b>8.2.1</b> Before QC</a></li>
<li class="chapter" data-level="8.2.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc"><i class="fa fa-check"></i><b>8.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="expression-overview.html"><a href="expression-overview.html#pca-plot"><i class="fa fa-check"></i><b>8.3</b> PCA plot</a><ul>
<li class="chapter" data-level="8.3.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-1"><i class="fa fa-check"></i><b>8.3.1</b> Before QC</a></li>
<li class="chapter" data-level="8.3.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-1"><i class="fa fa-check"></i><b>8.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="expression-overview.html"><a href="expression-overview.html#diffusion-map"><i class="fa fa-check"></i><b>8.4</b> Diffusion map</a><ul>
<li class="chapter" data-level="8.4.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-2"><i class="fa fa-check"></i><b>8.4.1</b> Before QC</a></li>
<li class="chapter" data-level="8.4.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-2"><i class="fa fa-check"></i><b>8.4.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="expression-overview.html"><a href="expression-overview.html#tsne-map"><i class="fa fa-check"></i><b>8.5</b> tSNE map</a><ul>
<li class="chapter" data-level="8.5.1" data-path="expression-overview.html"><a href="expression-overview.html#before-qc-3"><i class="fa fa-check"></i><b>8.5.1</b> Before QC</a></li>
<li class="chapter" data-level="8.5.2" data-path="expression-overview.html"><a href="expression-overview.html#after-qc-3"><i class="fa fa-check"></i><b>8.5.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="expression-overview.html"><a href="expression-overview.html#exercise-1"><i class="fa fa-check"></i><b>8.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-overview-reads.html"><a href="expression-overview-reads.html"><i class="fa fa-check"></i><b>9</b> Expression overview (Reads)</a></li>
<li class="chapter" data-level="10" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>10</b> Confounding factors</a><ul>
<li class="chapter" data-level="10.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>10.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="10.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>10.2.1</b> Detected genes</a></li>
<li class="chapter" data-level="10.2.2" data-path="confounding-factors.html"><a href="confounding-factors.html#control-erccs"><i class="fa fa-check"></i><b>10.2.2</b> Control ERCCs</a></li>
<li class="chapter" data-level="10.2.3" data-path="confounding-factors.html"><a href="confounding-factors.html#control-mts"><i class="fa fa-check"></i><b>10.2.3</b> Control MTs</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>10.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="10.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>10.4</b> Other confounders</a></li>
<li class="chapter" data-level="10.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise-2"><i class="fa fa-check"></i><b>10.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>11</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="12" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>12</b> Normalization for library size</a><ul>
<li class="chapter" data-level="12.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>12.2</b> Library size</a></li>
<li class="chapter" data-level="12.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>12.3</b> Normalisations</a><ul>
<li class="chapter" data-level="12.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>12.3.1</b> Raw</a></li>
<li class="chapter" data-level="12.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>12.3.2</b> CPM</a></li>
<li class="chapter" data-level="12.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#log2cpm"><i class="fa fa-check"></i><b>12.3.3</b> log2(CPM)</a></li>
<li class="chapter" data-level="12.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>12.3.4</b> TMM</a></li>
<li class="chapter" data-level="12.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#rle"><i class="fa fa-check"></i><b>12.3.5</b> RLE</a></li>
<li class="chapter" data-level="12.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>12.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#interpretation"><i class="fa fa-check"></i><b>12.4</b> Interpretation</a></li>
<li class="chapter" data-level="12.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#other-methods"><i class="fa fa-check"></i><b>12.5</b> Other methods</a></li>
<li class="chapter" data-level="12.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#visualize-genes"><i class="fa fa-check"></i><b>12.6</b> Visualize genes</a><ul>
<li class="chapter" data-level="12.6.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw-1"><i class="fa fa-check"></i><b>12.6.1</b> Raw</a></li>
<li class="chapter" data-level="12.6.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm-1"><i class="fa fa-check"></i><b>12.6.2</b> CPM</a></li>
<li class="chapter" data-level="12.6.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#log2cpm-1"><i class="fa fa-check"></i><b>12.6.3</b> log2(CPM)</a></li>
<li class="chapter" data-level="12.6.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile-1"><i class="fa fa-check"></i><b>12.6.4</b> Upperquantile</a></li>
<li class="chapter" data-level="12.6.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#fpkm"><i class="fa fa-check"></i><b>12.6.5</b> FPKM</a></li>
<li class="chapter" data-level="12.6.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tpm"><i class="fa fa-check"></i><b>12.6.6</b> TPM</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-3"><i class="fa fa-check"></i><b>12.7</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size (Reads)</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single-cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="expression-qc-umi" class="section level1">
<h1><span class="header-section-number">6</span> Expression QC (UMI)</h1>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">6.1</span> Introduction</h2>
<p>Once gene expression has been quantified the resulting expression matrix must be examined for poor quality cells which were not detected from QC of the raw reads. Failure to remove low quality cells at this stage will add a large amount of technical noise which will obscure the biological signals of interest in downstream analysis. We will demonstrate some simple filters that can be applied directly to the gene expression matrix with minimal extrinsic information.</p>
<p>Typically, the results are summarized using an <strong>expression matrix</strong>. In the expression matrix, each row represents a gene and each column represents a cell.</p>
<p>To illustrate cell QC, we consider a <a href="http://jdblischak.github.io/singleCellSeq/analysis/">dataset</a> from lymphoblast and induced pluripotent stem cells generated by John Blischak in <a href="http://giladlab.uchicago.edu/">Yoav Gilad</a>’s lab at the University of Chicago. The experiments were carried out on the Fluidigm C1 platform and to facilitate the quantification both unique molecular identifiers (UMIs) and ERCC <em>spike-ins</em> were used. For our purposes you need to download the files <code>annotation.txt</code>, <code>molecules.txt</code>, and <code>reads.txt</code> from <a href="https://github.com/hemberg-lab/scRNA.seq.course/tree/master/inst/materials/blischak">here</a> into the <code>blischak</code> folder in your working directory. These files are the copies of the original files made on the 15/03/16. We will use these copies for reproducibility purposes.</p>
<p>To illustrate how one can standardise the analysis of scRNA-seq data we will also be using the <a href="https://github.com/davismcc/scater">scater</a> package, described in the previous chapter.</p>
</div>
<div id="blischak-dataset" class="section level2">
<h2><span class="header-section-number">6.2</span> Blischak dataset</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scater, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)
<span class="kw">library</span>(knitr)
<span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
opts_chunk$<span class="kw">set</span>(<span class="dt">out.width=</span><span class="st">&#39;90%&#39;</span>, <span class="dt">fig.align =</span> <span class="st">&#39;center&#39;</span>)</code></pre></div>
<p>Load the data and annotations:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">molecules &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;blischak/molecules.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
anno &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;blischak/annotation.txt&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>How does the data look like?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
    <span class="kw">head</span>(molecules[ , <span class="dv">1</span>:<span class="dv">3</span>]), <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
    <span class="dt">caption =</span> <span class="st">&#39;A table of the first 6 rows and 3 columns of the molecules table.&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-18">Table 6.1: </span>A table of the first 6 rows and 3 columns of the molecules table.</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">NA19098.r1.A01</th>
<th align="right">NA19098.r1.A02</th>
<th align="right">NA19098.r1.A03</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000237683</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000187634</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000188976</td>
<td align="right">3</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td>ENSG00000187961</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td>ENSG00000187583</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td>ENSG00000187642</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
    <span class="kw">head</span>(anno), <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
    <span class="dt">caption =</span> <span class="st">&#39;A table of the first 6 rows of the anno table.&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-18">Table 6.2: </span>A table of the first 6 rows of the anno table.</caption>
<thead>
<tr class="header">
<th align="left">individual</th>
<th align="left">replicate</th>
<th align="left">well</th>
<th align="left">batch</th>
<th align="left">sample_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A01</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A01</td>
</tr>
<tr class="even">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A02</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A02</td>
</tr>
<tr class="odd">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A03</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A03</td>
</tr>
<tr class="even">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A04</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A04</td>
</tr>
<tr class="odd">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A05</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A05</td>
</tr>
<tr class="even">
<td align="left">NA19098</td>
<td align="left">r1</td>
<td align="left">A06</td>
<td align="left">NA19098.r1</td>
<td align="left">NA19098.r1.A06</td>
</tr>
</tbody>
</table>
<p>The data consists of 3 individuals and 3 replicates and therefore has 9 batches in total.</p>
<p>Let’s standardise the analysis by using the scater package described above. First, make scater SCESet classes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pheno_data &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, anno)
<span class="kw">rownames</span>(pheno_data) &lt;-<span class="st"> </span>pheno_data$sample_id
umi &lt;-<span class="st"> </span>scater::<span class="kw">newSCESet</span>(
    <span class="dt">countData =</span> molecules,
    <span class="dt">phenoData =</span> pheno_data
)</code></pre></div>
<p>Remove genes that are not expressed in any cell:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep_feature &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">is_exprs</span>(umi)) &gt;<span class="st"> </span><span class="dv">0</span>
umi &lt;-<span class="st"> </span>umi[keep_feature, ]</code></pre></div>
<p>Define control features (genes) - ERCC spike-ins and mitochondrial genes (<a href="http://jdblischak.github.io/singleCellSeq/analysis/qc-filter-ipsc.html">provided</a> by the authors):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ercc &lt;-<span class="st"> </span><span class="kw">featureNames</span>(umi)[<span class="kw">grepl</span>(<span class="st">&quot;ERCC-&quot;</span>, <span class="kw">featureNames</span>(umi))]
mt &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ENSG00000198899&quot;</span>, <span class="st">&quot;ENSG00000198727&quot;</span>, <span class="st">&quot;ENSG00000198888&quot;</span>,
        <span class="st">&quot;ENSG00000198886&quot;</span>, <span class="st">&quot;ENSG00000212907&quot;</span>, <span class="st">&quot;ENSG00000198786&quot;</span>,
        <span class="st">&quot;ENSG00000198695&quot;</span>, <span class="st">&quot;ENSG00000198712&quot;</span>, <span class="st">&quot;ENSG00000198804&quot;</span>,
        <span class="st">&quot;ENSG00000198763&quot;</span>, <span class="st">&quot;ENSG00000228253&quot;</span>, <span class="st">&quot;ENSG00000198938&quot;</span>,
        <span class="st">&quot;ENSG00000198840&quot;</span>)</code></pre></div>
<p>Calculate the quality metrics:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi &lt;-<span class="st"> </span>scater::<span class="kw">calculateQCMetrics</span>(
    umi,
    <span class="dt">feature_controls =</span> <span class="kw">list</span>(<span class="dt">ERCC =</span> ercc, <span class="dt">MT =</span> mt)
)</code></pre></div>
</div>
<div id="gene-qc" class="section level2">
<h2><span class="header-section-number">6.3</span> Gene QC</h2>
<div id="gene-dropouts" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Gene dropouts</h3>
<p>First, one can look at the gene expression frequency versus mean expression level to assess the effects of technical dropout in the dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotQC</span>(umi, <span class="dt">type =</span> <span class="st">&quot;exprs-freq-vs-mean&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:dropout-overview"></span>
<img src="scRNA-seq-course_files/figure-html/dropout-overview-1.png" alt="Dropout rate vs mean expression" width="90%" />
<p class="caption">
Figure 6.1: Dropout rate vs mean expression
</p>
</div>
<p>Interestingly, only less than half of the genes are expressed in more than 50% of the cells.</p>
</div>
<div id="gene-expression" class="section level3">
<h3><span class="header-section-number">6.3.2</span> Gene expression</h3>
<p>One can also look at the number of reads consumed by the top 50 expressed genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotQC</span>(umi, <span class="dt">type =</span> <span class="st">&quot;highest-expression&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:top50-gene-expr"></span>
<img src="scRNA-seq-course_files/figure-html/top50-gene-expr-1.png" alt="Number of total counts consumed by the top 50 expressed genes" width="90%" />
<p class="caption">
Figure 6.2: Number of total counts consumed by the top 50 expressed genes
</p>
</div>
<p>Not surprisingly, most of the very top expressed genes are either ERCCs or MT genes (Feature controls).</p>
</div>
</div>
<div id="gene-filtering" class="section level2">
<h2><span class="header-section-number">6.4</span> Gene filtering</h2>
<p>We can remove genes whose expression level is considered <strong>“undetectable”</strong>. We define a detectable gene expression level, in which at least 2 cells contain more than 1 read mapping to the gene.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_genes &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">counts</span>(umi), <span class="dv">1</span>, function(x) <span class="kw">length</span>(x[x &gt;<span class="st"> </span><span class="dv">1</span>]) &gt;=<span class="st"> </span><span class="dv">2</span>)
<span class="kw">fData</span>(umi)$use &lt;-<span class="st"> </span>filter_genes</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
    <span class="kw">as.data.frame</span>(<span class="kw">table</span>(filter_genes)),
    <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
    <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
    <span class="dt">caption =</span> <span class="st">&#39;The number of genes removed by gene filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-24">Table 6.3: </span>The number of genes removed by gene filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">filter_genes</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">4512</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">14214</td>
</tr>
</tbody>
</table>
<p>Depending on the cell-type, protocol and sequencing depth, other cut-offs may be appropriate.</p>
</div>
<div id="cell-qc" class="section level2">
<h2><span class="header-section-number">6.5</span> Cell QC</h2>
<div id="library-size" class="section level3">
<h3><span class="header-section-number">6.5.1</span> Library size</h3>
<p>Next we will look at the total number of RNA molecules detected per sample (if we were using read counts rather than UMI counts this would be total reads). Wells with few reads/molecules are likely to have been broken or failed to capture a cell, thus should be removed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(
    umi$total_counts,
    <span class="dt">breaks =</span> <span class="dv">100</span>
)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:total-counts-hist"></span>
<img src="scRNA-seq-course_files/figure-html/total-counts-hist-1.png" alt="Histogram of library sizes for all cells" width="90%" />
<p class="caption">
Figure 6.3: Histogram of library sizes for all cells
</p>
</div>
<p><strong>Exercise</strong></p>
<p>Apply a suitable filter to remove the cells that contain too few molecules. What distribution do you expect that the total number of molecules for each cell should follow?</p>
<p><strong>Answer</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_by_total_counts &lt;-<span class="st"> </span>(umi$total_counts &gt;<span class="st"> </span><span class="dv">10000</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
    <span class="kw">as.data.frame</span>(<span class="kw">table</span>(filter_by_total_counts)),
    <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
    <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
    <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by total counts filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-26">Table 6.4: </span>The number of cells removed by total counts filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">filter_by_total_counts</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">862</td>
</tr>
</tbody>
</table>
</div>
<div id="detected-genes-1" class="section level3">
<h3><span class="header-section-number">6.5.2</span> Detected genes (1)</h3>
<p>One of the simplest measures of cell quality is the number of genes that were detected. Cells with few detected genes may have been broken or dead prior to capture.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(
    umi$total_features,
    <span class="dt">breaks =</span> <span class="dv">100</span>
)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:total-features-hist"></span>
<img src="scRNA-seq-course_files/figure-html/total-features-hist-1.png" alt="Histogram of the number of detected genes in all cells" width="90%" />
<p class="caption">
Figure 6.4: Histogram of the number of detected genes in all cells
</p>
</div>
<p>Here we see that most cells have between 5,000-11,000 detected genes, which is normal for high-depth scRNA-seq. However this varies by experimental protocol and sequencing depth with droplet-based methods or lower sequencing-depth detecting fewer genes per cell. The feature to note is the <strong>“heavy tail”</strong> of the left hand side of the distribution. If detection rates were equal across the cells then the distribution should be approximately normal. Thus we remove those cells in the tail of the distribution (&lt;5,000 detected genes).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_by_expr_features &lt;-<span class="st"> </span>(umi$total_features &gt;<span class="st"> </span><span class="dv">5000</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
    <span class="kw">as.data.frame</span>(<span class="kw">table</span>(filter_by_expr_features)),
    <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
    <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
    <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by total features filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-28">Table 6.5: </span>The number of cells removed by total features filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">filter_by_expr_features</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">16</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">848</td>
</tr>
</tbody>
</table>
</div>
<div id="detected-genes-2" class="section level3">
<h3><span class="header-section-number">6.5.3</span> Detected genes (2)</h3>
<p>Additionally, we can plot the library size as a function of the number of the detected genes. This will also allow us to find other cell outliers.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPhenoData</span>(
    umi,
    <span class="kw">aes</span>(<span class="dt">x =</span> total_features, <span class="dt">y =</span> <span class="kw">log10</span>(total_counts), <span class="dt">colour =</span> batch)
)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:total-features-vs-counts"></span>
<img src="scRNA-seq-course_files/figure-html/total-features-vs-counts-1.png" alt="Library size vs number of detected genes" width="90%" />
<p class="caption">
Figure 6.5: Library size vs number of detected genes
</p>
</div>
</div>
<div id="erccs-and-mts" class="section level3">
<h3><span class="header-section-number">6.5.4</span> ERCCs and MTs</h3>
<p>Another measures of cell quality is the ration between ERCC <em>spike-in</em> RNAs and endogenous RNAs. This can be used to estimate the total amount of RNA in the captured cells. Cells with a high level of <em>spike-in</em> RNAs had low starting amounts of RNA, likely due to the cell being dead or stressed, or the RNA being degraded.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPhenoData</span>(
    umi,
    <span class="kw">aes_string</span>(<span class="dt">x =</span> <span class="st">&quot;total_features&quot;</span>,
               <span class="dt">y =</span> <span class="st">&quot;pct_counts_feature_controls_MT&quot;</span>,
               <span class="dt">colour =</span> <span class="st">&quot;batch&quot;</span>)
)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:mt-vs-counts"></span>
<img src="scRNA-seq-course_files/figure-html/mt-vs-counts-1.png" alt="Percentage of counts in MT genes" width="90%" />
<p class="caption">
Figure 6.6: Percentage of counts in MT genes
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPhenoData</span>(
    umi,
    <span class="kw">aes_string</span>(<span class="dt">x =</span> <span class="st">&quot;total_features&quot;</span>,
               <span class="dt">y =</span> <span class="st">&quot;pct_counts_feature_controls_ERCC&quot;</span>,
               <span class="dt">colour =</span> <span class="st">&quot;batch&quot;</span>)
)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:ercc-vs-counts"></span>
<img src="scRNA-seq-course_files/figure-html/ercc-vs-counts-1.png" alt="Percentage of counts in ERCCs" width="90%" />
<p class="caption">
Figure 6.7: Percentage of counts in ERCCs
</p>
</div>
<p>This analysis shows that majority of the cells from NA19098.r2 batch have a very high ERCC/Endo ratio. Indeed, it has been shown by the authors that this batch contains cells of smaller size.</p>
<p><strong>Exercise</strong></p>
<p>Create filters for removing batch NA19098.r2 and cells with high expression of mitochondrial genes (&gt;10% of total counts in a cell).</p>
<p><strong>Answer</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_by_ERCC &lt;-<span class="st"> </span>umi$batch !=<span class="st"> &quot;NA19098.r2&quot;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">as.data.frame</span>(<span class="kw">table</span>(filter_by_ERCC)),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by ERCC filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-30">Table 6.6: </span>The number of cells removed by ERCC filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">filter_by_ERCC</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">96</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">768</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter_by_MT &lt;-<span class="st"> </span>umi$pct_counts_feature_controls_MT &lt;<span class="st"> </span><span class="dv">10</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">as.data.frame</span>(<span class="kw">table</span>(filter_by_MT)),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by MT filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-32">Table 6.7: </span>The number of cells removed by MT filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">filter_by_MT</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">31</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">833</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cell-filtering" class="section level2">
<h2><span class="header-section-number">6.6</span> Cell filtering</h2>
<div id="default" class="section level3">
<h3><span class="header-section-number">6.6.1</span> Default</h3>
<p>To filter out outlier cells one could use the defaults provided by the scater package:</p>
<ul>
<li><p><strong>filter_on_total_features</strong>: filter out cells based on their total_features being (by default) more than 5 median absolute deviations from the median total_features for the dataset</p></li>
<li><p><strong>filter_on_total_counts</strong>: filter out cells based on their log10-total_counts being (by default) more than 5 median absolute deviations from the median log10-total_counts for the dataset?</p></li>
<li><p><strong>filter_on_pct_counts_feature_controls</strong>: filter out cells on the basis that the percentage of counts from feature controls is higher than a defined threhold (default is 80%)?</p></li>
<li><p><strong>is_cell_control</strong>: filter out cells that were defined as controls</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi$use_default &lt;-<span class="st"> </span>(
    <span class="co"># remove cells with unusual numbers of genes</span>
    !umi$filter_on_total_features &amp;
<span class="st">    </span><span class="co"># sufficient molecules counted</span>
<span class="st">    </span>!umi$filter_on_total_counts &amp;
<span class="st">    </span><span class="co"># sufficient endogenous RNA</span>
<span class="st">    </span>!umi$filter_on_pct_counts_feature_controls_ERCC &amp;
<span class="st">    </span><span class="co"># remove cells with unusual number of reads in MT genes</span>
<span class="st">    </span>!umi$filter_on_pct_counts_feature_controls_MT &amp;
<span class="st">    </span><span class="co"># controls shouldn&#39;t be used in downstream analysis</span>
<span class="st">    </span>!umi$is_cell_control
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">as.data.frame</span>(<span class="kw">table</span>(umi$use_default)),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by default filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-34">Table 6.8: </span>The number of cells removed by default filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">858</td>
</tr>
</tbody>
</table>
<p>This filtering can be performed even without looking at the plots above.</p>
</div>
<div id="automatic" class="section level3">
<h3><span class="header-section-number">6.6.2</span> Automatic</h3>
<p>Another option available in <strong>scater</strong> is to conduct PCA on a set of QC metrics. The advantage of doing this is that the QC metrics focus on technical aspects of the libraries that are likely to distinguish problematics cells. Automatic outlier detection on PCA plots using QC metrics is available to help identify potentially problematic cells.</p>
<p>By default, the following metrics are used for PCA-based outlier detection:</p>
<ul>
<li><strong>pct_counts_top_100_features</strong></li>
<li><strong>total_features</strong></li>
<li><strong>pct_counts_feature_controls</strong></li>
<li><strong>n_detected_feature_controls</strong></li>
<li><strong>log10_counts_endogenous_features</strong></li>
<li><strong>log10_counts_feature_controls</strong></li>
</ul>
<p>A particular set of variables to be used can be specified with the selected_variables argument as shown in the example below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi &lt;-
scater::<span class="kw">plotPCA</span>(umi,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>, 
                <span class="dt">shape_by =</span> <span class="st">&quot;filter_on_total_features&quot;</span>,
                <span class="dt">pca_data_input =</span> <span class="st">&quot;pdata&quot;</span>,
                <span class="dt">detect_outliers =</span> <span class="ot">TRUE</span>,
                <span class="dt">return_SCESet =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## The following cells/samples are detected as outliers:
## NA19098.r2.A01
## NA19098.r2.A02
## NA19098.r2.A06
## NA19098.r2.A09
## NA19098.r2.A10
## NA19098.r2.A12
## NA19098.r2.B01
## NA19098.r2.B03
## NA19098.r2.B04
## NA19098.r2.B05
## NA19098.r2.B07
## NA19098.r2.B11
## NA19098.r2.B12
## NA19098.r2.C01
## NA19098.r2.C02
## NA19098.r2.C03
## NA19098.r2.C04
## NA19098.r2.C05
## NA19098.r2.C06
## NA19098.r2.C07
## NA19098.r2.C08
## NA19098.r2.C09
## NA19098.r2.C10
## NA19098.r2.C11
## NA19098.r2.C12
## NA19098.r2.D01
## NA19098.r2.D02
## NA19098.r2.D03
## NA19098.r2.D04
## NA19098.r2.D07
## NA19098.r2.D08
## NA19098.r2.D09
## NA19098.r2.D10
## NA19098.r2.D12
## NA19098.r2.E01
## NA19098.r2.E02
## NA19098.r2.E03
## NA19098.r2.E04
## NA19098.r2.E05
## NA19098.r2.E06
## NA19098.r2.E07
## NA19098.r2.E12
## NA19098.r2.F01
## NA19098.r2.F02
## NA19098.r2.F07
## NA19098.r2.F08
## NA19098.r2.F09
## NA19098.r2.F10
## NA19098.r2.F11
## NA19098.r2.F12
## NA19098.r2.G01
## NA19098.r2.G02
## NA19098.r2.G03
## NA19098.r2.G05
## NA19098.r2.G06
## NA19098.r2.G08
## NA19098.r2.G09
## NA19098.r2.G10
## NA19098.r2.G11
## NA19098.r2.H01
## NA19098.r2.H02
## NA19098.r2.H03
## NA19098.r2.H04
## NA19098.r2.H05
## NA19098.r2.H06
## NA19098.r2.H07
## NA19098.r2.H08
## NA19098.r2.H10
## NA19098.r2.H12
## NA19101.r3.A02
## NA19101.r3.C12
## NA19101.r3.D01
## NA19101.r3.E08
## Variables with highest loadings for PC1 and PC2:
## 
##                                            PC1         PC2
## ---------------------------------  -----------  ----------
## pct_counts_top_100_features          0.4771343   0.3009332
## pct_counts_feature_controls          0.4735839   0.3309562
## n_detected_feature_controls          0.1332811   0.5367629
## log10_counts_feature_controls       -0.1427373   0.5911762
## total_features                      -0.5016681   0.2936705
## log10_counts_endogenous_features    -0.5081855   0.2757918</code></pre>
<div class="figure" style="text-align: center"><span id="fig:auto-cell-filt"></span>
<img src="scRNA-seq-course_files/figure-html/auto-cell-filt-1.png" alt="PCA plot used for automatic detection of cell outliers" width="90%" />
<p class="caption">
Figure 6.8: PCA plot used for automatic detection of cell outliers
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">as.data.frame</span>(<span class="kw">table</span>(umi$outlier)),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by automatic filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-35">Table 6.9: </span>The number of cells removed by automatic filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">791</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">73</td>
</tr>
</tbody>
</table>
</div>
<div id="manual" class="section level3">
<h3><span class="header-section-number">6.6.3</span> Manual</h3>
<p>However, since we performed a more detailed analysis, visualized different features and defined our own filters we can use them instead of the default ones:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi$use &lt;-<span class="st"> </span>(
    <span class="co"># sufficient features (genes)</span>
    filter_by_expr_features &amp;
<span class="st">    </span><span class="co"># sufficient molecules counted</span>
<span class="st">    </span>filter_by_total_counts &amp;
<span class="st">    </span><span class="co"># sufficient endogenous RNA</span>
<span class="st">    </span>filter_by_ERCC &amp;
<span class="st">    </span><span class="co"># remove cells with unusual number of reads in MT genes</span>
<span class="st">    </span>filter_by_MT
)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr::<span class="kw">kable</span>(
  <span class="kw">as.data.frame</span>(<span class="kw">table</span>(umi$use)),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>,
  <span class="dt">row.names =</span> <span class="ot">FALSE</span>,
  <span class="dt">caption =</span> <span class="st">&#39;The number of cells removed by manual filter (FALSE)&#39;</span>
)</code></pre></div>
<table>
<caption><span id="tab:unnamed-chunk-37">Table 6.10: </span>The number of cells removed by manual filter (FALSE)</caption>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">141</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">723</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compare-filterings" class="section level2">
<h2><span class="header-section-number">6.7</span> Compare filterings</h2>
<p><strong>Exercise</strong></p>
<p>Compare the default, automatic and manual cell filters. Plot a Venn diagram of the outlier cells from these filterings.</p>
<p><strong>Hint</strong>: Use <code>limma::vennCounts</code> and <code>limma::vennDiagram</code> functions from the <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">limma</a> package to make a Venn diagram.</p>
<p><strong>Answer</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">def &lt;-<span class="st"> </span><span class="kw">colnames</span>(umi)[!umi$use_default]
auto &lt;-<span class="st"> </span><span class="kw">colnames</span>(umi)[umi$outlier]
man &lt;-<span class="st"> </span><span class="kw">colnames</span>(umi)[!umi$use]
venn.diag &lt;-<span class="st"> </span>limma::<span class="kw">vennCounts</span>(<span class="kw">cbind</span>(<span class="kw">colnames</span>(umi) %in%<span class="st"> </span>def,
                                     <span class="kw">colnames</span>(umi) %in%<span class="st"> </span>auto,
                                     <span class="kw">colnames</span>(umi) %in%<span class="st"> </span>man))
limma::<span class="kw">vennDiagram</span>(venn.diag,
                   <span class="dt">names =</span> <span class="kw">c</span>(<span class="st">&quot;Default&quot;</span>, <span class="st">&quot;Automatic&quot;</span>, <span class="st">&quot;Manual&quot;</span>),
                   <span class="dt">circle.col =</span> <span class="kw">c</span>(<span class="st">&quot;magenta&quot;</span>, <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:cell-filt-comp"></span>
<img src="scRNA-seq-course_files/figure-html/cell-filt-comp-1.png" alt="Comparison of the default, automatic and manual cell filters" width="90%" />
<p class="caption">
Figure 6.9: Comparison of the default, automatic and manual cell filters
</p>
</div>
</div>
<div id="save-the-data" class="section level2">
<h2><span class="header-section-number">6.8</span> Save the data</h2>
<p>Dimensions of the QCed dataset (do not forget about the gene filter we defined above):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(umi[<span class="kw">fData</span>(umi)$use, <span class="kw">pData</span>(umi)$use])</code></pre></div>
<pre><code>## Features  Samples 
##    14214      723</code></pre>
<p>Save the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">saveRDS</span>(umi, <span class="dt">file =</span> <span class="st">&quot;blischak/umi.rds&quot;</span>)</code></pre></div>
</div>
<div id="exercise" class="section level2">
<h2><span class="header-section-number">6.9</span> Exercise</h2>
<p>Perform exactly the same QC analysis with read counts of the same Blischak data. Use <code>blischak/reads.txt</code> file to load the reads. Once you have finished please compare your results to ours (next chapter).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="scater-package.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="expression-qc-reads.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/wikiselev/bookdown-demo/edit/master/05-exprs-qc.Rmd",
"text": "Edit"
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>


</body>

</html>
