<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single-cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single-cell RNA-seq data">
  <meta name="generator" content="bookdown 0.0.73 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of single-cell RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of single-cell RNA-seq data" />
  
  
  

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews and Martin Hemberg">

<meta name="date" content="2016-06-16">

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="confounding-factors-reads.html">
<link rel="next" href="normalization-for-library-size-reads.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.1</b> Registration</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.2</b> GitHub</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.4</b> Prerequisites</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> Spike-ins</a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a><ul>
<li class="chapter" data-level="4.5.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#umi-quantification"><i class="fa fa-check"></i><b>4.5.1</b> UMI quantification</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>5</b> scater package</a><ul>
<li class="chapter" data-level="5.1" data-path="scater-package.html"><a href="scater-package.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>5.2</b> scater workflow</a></li>
<li class="chapter" data-level="5.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>5.3</b> Terminology</a></li>
<li class="chapter" data-level="5.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>5.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>6</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>6.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>6.3</b> Cell QC</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>6.3.1</b> Library size</a></li>
<li class="chapter" data-level="6.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>6.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="6.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>6.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>6.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="6.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>6.4.1</b> Default Thresholds</a></li>
<li class="chapter" data-level="6.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>6.4.2</b> Automatic</a></li>
<li class="chapter" data-level="6.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>6.4.3</b> Manual</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>6.5</b> Compare filterings</a></li>
<li class="chapter" data-level="6.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>6.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="6.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>6.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="6.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>6.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>6.7</b> Save the data</a></li>
<li class="chapter" data-level="6.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>6.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>7</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="8" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>8</b> Data visualization</a><ul>
<li class="chapter" data-level="8.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>8.2</b> PCA plot</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>8.2.1</b> Before QC</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>8.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>8.3</b> tSNE map</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>8.3.1</b> Before QC</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>8.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>8.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-overview-reads.html"><a href="expression-overview-reads.html"><i class="fa fa-check"></i><b>9</b> Expression overview (Reads)</a></li>
<li class="chapter" data-level="10" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>10</b> Confounding factors</a><ul>
<li class="chapter" data-level="10.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>10.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="10.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>10.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>10.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="10.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>10.4</b> Other confounders</a></li>
<li class="chapter" data-level="10.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>10.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>11</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="12" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>12</b> Normalization for library size</a><ul>
<li class="chapter" data-level="12.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>12.2</b> Library size</a></li>
<li class="chapter" data-level="12.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>12.3</b> Normalisations</a><ul>
<li class="chapter" data-level="12.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>12.3.1</b> Raw</a></li>
<li class="chapter" data-level="12.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>12.3.2</b> CPM</a></li>
<li class="chapter" data-level="12.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>12.3.3</b> TMM</a></li>
<li class="chapter" data-level="12.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#lsf"><i class="fa fa-check"></i><b>12.3.4</b> LSF</a></li>
<li class="chapter" data-level="12.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>12.3.5</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="12.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>12.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#other-methods"><i class="fa fa-check"></i><b>12.4</b> Other methods</a></li>
<li class="chapter" data-level="12.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>12.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="14" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html"><i class="fa fa-check"></i><b>14</b> Remove confounders using controls</a><ul>
<li class="chapter" data-level="14.1" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#introduction-5"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#brennecke-method"><i class="fa fa-check"></i><b>14.2</b> Brennecke method</a></li>
<li class="chapter" data-level="14.3" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>14.3</b> Remove Unwanted Variation</a></li>
<li class="chapter" data-level="14.4" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-1"><i class="fa fa-check"></i><b>14.4</b> Effectiveness 1</a></li>
<li class="chapter" data-level="14.5" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-2"><i class="fa fa-check"></i><b>14.5</b> Effectiveness 2</a></li>
<li class="chapter" data-level="14.6" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#exercise-2"><i class="fa fa-check"></i><b>14.6</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="remove-confounders-using-controls-reads.html"><a href="remove-confounders-using-controls-reads.html"><i class="fa fa-check"></i><b>15</b> Remove confounders using controls (Reads)</a></li>
<li class="chapter" data-level="16" data-path="clustering-analysis.html"><a href="clustering-analysis.html"><i class="fa fa-check"></i><b>16</b> Clustering analysis</a><ul>
<li class="chapter" data-level="16.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#introduction-6"><i class="fa fa-check"></i><b>16.1</b> Introduction</a></li>
<li class="chapter" data-level="16.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#dimensionality-reductions"><i class="fa fa-check"></i><b>16.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="16.2.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pca"><i class="fa fa-check"></i><b>16.2.1</b> PCA</a></li>
<li class="chapter" data-level="16.2.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#spectral"><i class="fa fa-check"></i><b>16.2.2</b> Spectral</a></li>
<li class="chapter" data-level="16.2.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne"><i class="fa fa-check"></i><b>16.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#clustering-methods"><i class="fa fa-check"></i><b>16.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="16.3.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>16.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="16.3.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#k-means"><i class="fa fa-check"></i><b>16.3.2</b> k-means</a></li>
<li class="chapter" data-level="16.3.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#graph-based-methods"><i class="fa fa-check"></i><b>16.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#challenges-in-clustering"><i class="fa fa-check"></i><b>16.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="16.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>16.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="16.5.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sincera"><i class="fa fa-check"></i><b>16.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="16.5.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pcareduce"><i class="fa fa-check"></i><b>16.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="16.5.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sc3"><i class="fa fa-check"></i><b>16.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="16.5.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne-k-means"><i class="fa fa-check"></i><b>16.5.4</b> tSNE + <em>k</em>-means</a></li>
<li class="chapter" data-level="16.5.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#snn-cliq"><i class="fa fa-check"></i><b>16.5.5</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>17</b> Clustering analysis</a><ul>
<li class="chapter" data-level="17.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>17.1</b> Patient dataset</a></li>
<li class="chapter" data-level="17.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>17.2</b> SC3</a></li>
<li class="chapter" data-level="17.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>17.3</b> pcaReduce</a></li>
<li class="chapter" data-level="17.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>17.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="17.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>17.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="17.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>17.6</b> SINCERA</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html"><i class="fa fa-check"></i><b>18</b> DE expression in a real dataset</a><ul>
<li class="chapter" data-level="18.1" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#fitting-the-models"><i class="fa fa-check"></i><b>18.1</b> Fitting the models</a></li>
<li class="chapter" data-level="18.2" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#the-michaelis-menten-equation"><i class="fa fa-check"></i><b>18.2</b> The Michaelis-Menten Equation</a></li>
<li class="chapter" data-level="18.3" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#right-outliers"><i class="fa fa-check"></i><b>18.3</b> Right outliers</a></li>
<li class="chapter" data-level="18.4" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#validation-of-de-results"><i class="fa fa-check"></i><b>18.4</b> Validation of DE results</a></li>
<li class="chapter" data-level="18.5" data-path="de-expression-in-a-real-dataset.html"><a href="de-expression-in-a-real-dataset.html#comparing-m3drop-to-other-methods"><i class="fa fa-check"></i><b>18.5</b> Comparing M3Drop to other methods</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="ordering-of-cells-according-to-pseudotime.html"><a href="ordering-of-cells-according-to-pseudotime.html"><i class="fa fa-check"></i><b>19</b> Ordering of cells according to pseudotime</a></li>
<li class="chapter" data-level="20" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>20</b> Introduction to differential expression (DE) analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#background-bulk-rna-seq"><i class="fa fa-check"></i><b>20.1</b> Background (bulk RNA-seq)</a></li>
<li class="chapter" data-level="20.2" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#de-analysis-for-scrna-seq"><i class="fa fa-check"></i><b>20.2</b> DE analysis for scRNA-seq</a></li>
<li class="chapter" data-level="20.3" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#synthetic-data-for-scrna-seq"><i class="fa fa-check"></i><b>20.3</b> Synthetic data for scRNA-seq</a></li>
<li class="chapter" data-level="20.4" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#modelled-expression-of-a-single-gene"><i class="fa fa-check"></i><b>20.4</b> Modelled expression of a single gene</a></li>
<li class="chapter" data-level="20.5" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#possible-shapes-of-the-poisson-beta-distribution"><i class="fa fa-check"></i><b>20.5</b> Possible shapes of the Poisson-Beta distribution</a></li>
<li class="chapter" data-level="20.6" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#effect-of-sampling-size-on-parameter-estimates"><i class="fa fa-check"></i><b>20.6</b> Effect of sampling size on parameter estimates</a></li>
<li class="chapter" data-level="20.7" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#drop-out-noise"><i class="fa fa-check"></i><b>20.7</b> Drop-out noise</a></li>
<li class="chapter" data-level="20.8" data-path="introduction-to-differential-expression-de-analysis.html"><a href="introduction-to-differential-expression-de-analysis.html#over--and-under-dispersion-noise"><i class="fa fa-check"></i><b>20.8</b> Over- and under-dispersion noise</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html"><i class="fa fa-check"></i><b>21</b> DE expression in a synthetic dataset</a><ul>
<li class="chapter" data-level="21.1" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#generation-of-a-synthetic-dataset"><i class="fa fa-check"></i><b>21.1</b> Generation of a synthetic dataset</a></li>
<li class="chapter" data-level="21.2" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#differential-expression-in-scrna-seq"><i class="fa fa-check"></i><b>21.2</b> Differential expression in scRNA-seq</a></li>
<li class="chapter" data-level="21.3" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>21.3</b> Kolmogorov-Smirnov test</a></li>
<li class="chapter" data-level="21.4" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#performance-of-the-ks-test"><i class="fa fa-check"></i><b>21.4</b> Performance of the KS test</a></li>
<li class="chapter" data-level="21.5" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#finding-de-genes-using-deseq2"><i class="fa fa-check"></i><b>21.5</b> Finding DE genes using DESeq2</a></li>
<li class="chapter" data-level="21.6" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#finding-de-genes-using-scde"><i class="fa fa-check"></i><b>21.6</b> Finding DE genes using SCDE</a></li>
<li class="chapter" data-level="21.7" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>21.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="21.8" data-path="de-expression-in-a-synthetic-dataset.html"><a href="de-expression-in-a-synthetic-dataset.html#beyond-changes-in-the-mean"><i class="fa fa-check"></i><b>21.8</b> Beyond changes in the mean</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html"><i class="fa fa-check"></i><b>22</b> DE expression in a real dataset</a><ul>
<li class="chapter" data-level="22.1" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#introduction-7"><i class="fa fa-check"></i><b>22.1</b> Introduction</a></li>
<li class="chapter" data-level="22.2" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#ks-test"><i class="fa fa-check"></i><b>22.2</b> KS-test</a></li>
<li class="chapter" data-level="22.3" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#deseq2"><i class="fa fa-check"></i><b>22.3</b> DESeq2</a></li>
<li class="chapter" data-level="22.4" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#scde"><i class="fa fa-check"></i><b>22.4</b> SCDE</a></li>
<li class="chapter" data-level="22.5" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#comparison-of-the-methods-1"><i class="fa fa-check"></i><b>22.5</b> Comparison of the methods</a></li>
<li class="chapter" data-level="22.6" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#visualisation-of-the-results"><i class="fa fa-check"></i><b>22.6</b> Visualisation of the results</a><ul>
<li class="chapter" data-level="22.6.1" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#volcano-plots"><i class="fa fa-check"></i><b>22.6.1</b> Volcano plots</a></li>
<li class="chapter" data-level="22.6.2" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#ma-plot"><i class="fa fa-check"></i><b>22.6.2</b> MA-plot</a></li>
<li class="chapter" data-level="22.6.3" data-path="de-expression-in-a-real-dataset-1.html"><a href="de-expression-in-a-real-dataset-1.html#heatmap-of-de-genes"><i class="fa fa-check"></i><b>22.6.3</b> Heatmap of DE genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>23</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single-cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="normalization-for-library-size" class="section level1">
<h1><span class="header-section-number">12</span> Normalization for library size</h1>
<div id="introduction-4" class="section level2">
<h2><span class="header-section-number">12.1</span> Introduction</h2>
<p>In the previous chapter we identified important confounding factors and explanatory variables. scater allows one to account for these variables in subsequent statistical models or to condition them out using <code>normaliseExprs()</code>, if so desired. This can be done by providing a design matrix to <code>normaliseExprs()</code>. We are not covering this topic here, but you can try to do it yourself as an exercise.</p>
<p>Instead we will explore how simple size-factor normalisations correcting for library size can remove the effects of some of the confounders and explanatory variables.</p>
</div>
<div id="library-size-1" class="section level2">
<h2><span class="header-section-number">12.2</span> Library size</h2>
<p>Library sizes vary because scRNA-seq data is often sequenced on highly multiplexed platforms the total reads which are derived from each cell may differ substantially. Some quantification methods (eg. <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a>, <a href="http://deweylab.github.io/RSEM/">RSEM</a>) incorporated library size when determining gene expression estimates thus do not require this normalization.</p>
<p>However, if another quantification method was used then library size must be corrected for by multiplying or dividing each column of the expression matrix by a “normalization factor” which is an estimate of the library size relative to the other cells. Many methods to correct for library size have been developped for bulk RNA-seq and can be equally applied to scRNA-seq (eg. UQ, SF, CPM, RPKM, FPKM, TPM).</p>
</div>
<div id="normalisations" class="section level2">
<h2><span class="header-section-number">12.3</span> Normalisations</h2>
<p>The simplest way to normalize this data is to convert it to counts per million (<strong>CPM</strong>) by dividing each column by its total then multiplying by 1,000,000. Note that spike-ins should be excluded from the calculation of total expression in order to correct for total cell RNA content, therefore we will only use endogenous genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">calc_cpm &lt;-
function (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) 
{
    norm_factor &lt;-<span class="st"> </span><span class="kw">colSums</span>(expr_mat[-spikes, ])
    <span class="kw">return</span>(<span class="kw">t</span>(<span class="kw">t</span>(expr_mat)/norm_factor)) *<span class="st"> </span><span class="dv">10</span>^<span class="dv">6</span>
}</code></pre></div>
<p>One potential drawback of <strong>CPM</strong> is if your sample contains genes that are both very highly expressed and differentially expressed across the cells. In this case, the total molecules in the cell may depend of whether such genes are on/off in the cell and normalizing by total molecules may hide the differential expression of those genes and/or falsely create differential expression for the remaining genes.</p>
<p><strong>Note</strong>: RPKM, FPKM and TPM are variants on CPM which further adjust counts by the length of the respective gene/transcript.</p>
<p>To deal with this potentiality several other measures were devised:</p>
<p>The <strong>size factor (SF)</strong> was proposed and popularized by DESeq (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-10-r106">Anders and Huber (2010)</a>). First the geometric mean of each gene across all cells is calculated. The size factor for each cell is the median across genes of the ratio of the expression to the gene’s geometric mean. A draw back to this method is that since it uses the geometric mean only genes with non-zero expression values across all cells can be used in its calculation, making it unadvisable for large low-depth scRNASeq experiments. edgeR &amp; scater call this method <strong>RLE</strong> for “relative log expression”.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">calc_sf &lt;-
function (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) 
{
    geomeans &lt;-<span class="st"> </span><span class="kw">exp</span>(<span class="kw">rowMeans</span>(<span class="kw">log</span>(expr_mat[-spikes, ])))
    SF &lt;-<span class="st"> </span>function(cnts) {
        <span class="kw">median</span>((cnts/geomeans)[(<span class="kw">is.finite</span>(geomeans) &amp;<span class="st"> </span>geomeans &gt;<span class="st"> </span>
<span class="st">            </span><span class="dv">0</span>)])
    }
    norm_factor &lt;-<span class="st"> </span><span class="kw">apply</span>(expr_mat[-spikes, ], <span class="dv">2</span>, SF)
    <span class="kw">return</span>(<span class="kw">t</span>(<span class="kw">t</span>(expr_mat)/norm_factor))
}</code></pre></div>
<p>The <strong>upperquartile (UQ)</strong> was proposed by <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-94">Bullard et al (2010)</a>. Here each column is divided by the 75% quantile of the counts for each library. Often the calculated quantile is scaled by the median across cells to keep the absolute level of expression relatively consistent. A draw back to this method is that for low-depth scRNASeq experiments the large number of undetected genes may result in the 75% quantile being zero (or close to it). This limitation can be overcome by generalizing the idea and using a higher quantile (eg. the 99% quantile is the default in scater) or by excluding zeros prior to calculating the 75% quantile.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">calc_uq &lt;-
function (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) 
{
    UQ &lt;-<span class="st"> </span>function(x) {
        <span class="kw">quantile</span>(x[x &gt;<span class="st"> </span><span class="dv">0</span>], <span class="fl">0.75</span>)
    }
    uq &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">apply</span>(expr_mat[-spikes, ], <span class="dv">2</span>, UQ))
    norm_factor &lt;-<span class="st"> </span>uq/<span class="kw">median</span>(uq)
    <span class="kw">return</span>(<span class="kw">t</span>(<span class="kw">t</span>(expr_mat)/norm_factor))
}</code></pre></div>
<p>Another method is called <strong>TMM</strong> is the weighted trimmed mean of M-values (to the reference) proposed by <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">Robinson and Oshlack (2010)</a>. The M-values in question are the gene-wise log2 fold changes between individual cells. One cell is used as the reference then the M-values for each other cell is calculated compared to this reference. These values are then trimmed by removing the top and bottom ~30%, and the average of the remaining values is calculated by weighting them to account for the effect of the log scale on variance. Each non-reference cell is multiplied by the calculated factor. Two potential issues with this method are insufficient non-zero genes left after trimming, and the assumption that most genes are not differentially expressed.</p>
<p>We will use visual inspection of PCA plots and calculation of cell-wise relative log expression (<code>calc_cell_RLE()</code>) to compare the efficiency of different normalization methods. Cells with many[few] reads have higher[lower] than median expression for most genes resulting in a positive[negative] RLE across the cell, whereas normalized cells have an RLE close to zero.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">calc_cell_RLE &lt;-
function (expr_mat, <span class="dt">spikes =</span> <span class="ot">NULL</span>) 
{
    RLE_gene &lt;-<span class="st"> </span>function(x) {
        if (<span class="kw">median</span>(<span class="kw">unlist</span>(x)) &gt;<span class="st"> </span><span class="dv">0</span>) {
            <span class="kw">log</span>((x +<span class="st"> </span><span class="dv">1</span>)/(<span class="kw">median</span>(<span class="kw">unlist</span>(x)) +<span class="st"> </span><span class="dv">1</span>))/<span class="kw">log</span>(<span class="dv">2</span>)
        }
        else {
            <span class="kw">rep</span>(<span class="ot">NA</span>, <span class="dt">times =</span> <span class="kw">length</span>(x))
        }
    }
    if (!<span class="kw">is.null</span>(spikes)) {
        RLE_matrix &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(expr_mat[-spikes, ], <span class="dv">1</span>, RLE_gene))
    }
    else {
        RLE_matrix &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">apply</span>(expr_mat, <span class="dv">1</span>, RLE_gene))
    }
    cell_RLE &lt;-<span class="st"> </span><span class="kw">apply</span>(RLE_matrix, <span class="dv">2</span>, median, <span class="dt">na.rm =</span> T)
    <span class="kw">return</span>(cell_RLE)
}</code></pre></div>
<p>The <strong>RLE</strong>, <strong>TMM</strong>, and <strong>UQ</strong> size-factor methods were developed for bulk RNA-seq data and, depending on the experimental context, may not be appropriate for single-cell RNA-seq data, as their underlying assumptions may be problematically violated. <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7">Lun et al (2016)</a> recently published a size-factor normalisation method specifically designed for scRNA-seq data and accounting for single-cell biases, which we will call <strong>LSF</strong> (Lun Sum Factors). Briefly, expression values are summed across pools of cells and the summed values are used to compute normalization size-factors per pool. The pool-based size factors can then be deconvolved into cell-specific size factors, which can be used for normalization in the same way as other size factors.</p>
<p><strong>scater</strong> acts as a wrapper for the <code>calcNormFactors</code> function from <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a> which implements several library size normalization methods making it easy to apply any of these methods to our data. The <strong>LSF</strong> method is implementated in the Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/scran.html">scran</a>, which allows seamless integrationinto the <code>scater</code> workflow. (The <code>scran</code> package itself depends on <code>scater</code>).</p>
<p><strong>Note:</strong> edgeR makes extra adjustments to some of the normalization methods which may result in somewhat different results than if the original methods are followed exactly, e.g. edgeR’s and scater’s “RLE” method which is based on the “size factor” used by <a href="http://bioconductor.org/packages/release/bioc/html/DESeq.html">DESeq</a> may give different results to the <code>estimateSizeFactorsForMatrix</code> method in the DESeq/DESeq2 packages. In addition, some versions of edgeR will not calculate the normalization factors correctly unless <code>lib.size</code> is set at 1 for all cells.</p>
<p>We will continue to work with the Blischak data that was used in the previous chapter.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scater, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)
<span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
umi &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;blischak/umi.rds&quot;</span>)
umi.qc &lt;-<span class="st"> </span>umi[<span class="kw">fData</span>(umi)$use, <span class="kw">pData</span>(umi)$use]
endog_genes &lt;-<span class="st"> </span>!<span class="kw">fData</span>(umi.qc)$is_feature_control</code></pre></div>
<div id="raw" class="section level3">
<h3><span class="header-section-number">12.3.1</span> Raw</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;counts&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-raw"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-raw-1.png" alt="PCA plot of the blischak data" width="90%" />
<p class="caption">
Figure 12.1: PCA plot of the blischak data
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">calc_cell_RLE</span>(<span class="kw">counts</span>(umi.qc)),
        <span class="dt">col =</span> <span class="st">&quot;grey50&quot;</span>,
        <span class="dt">ylab =</span> <span class="st">&quot;RLE&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;&quot;</span>, <span class="dt">ylim=</span><span class="kw">c</span>(-<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-ours-rle-raw"></span>
<img src="11-exprs-norm_files/figure-html/norm-ours-rle-raw-1.png" alt="Cell-wise RLE of the blischak data" width="90%" />
<p class="caption">
Figure 12.2: Cell-wise RLE of the blischak data
</p>
</div>
</div>
<div id="cpm" class="section level3">
<h3><span class="header-section-number">12.3.2</span> CPM</h3>
<p>scater performs this normalisation by default, you can control it by changing <code>exprs_values</code> parameter to <code>&quot;exprs&quot;</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;cpm&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-cpm"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-cpm-1.png" alt="PCA plot of the blischak data after CPM normalisation" width="90%" />
<p class="caption">
Figure 12.3: PCA plot of the blischak data after CPM normalisation
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">calc_cell_RLE</span>(<span class="kw">cpm</span>(umi.qc)),
        <span class="dt">col =</span> <span class="st">&quot;grey50&quot;</span>,
        <span class="dt">ylab =</span> <span class="st">&quot;RLE&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;&quot;</span>, <span class="dt">ylim =</span> <span class="kw">c</span>(-<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-ours-rle-cpm"></span>
<img src="11-exprs-norm_files/figure-html/norm-ours-rle-cpm-1.png" alt="Cell-wise RLE of the blischak data" width="90%" />
<p class="caption">
Figure 12.4: Cell-wise RLE of the blischak data
</p>
</div>
</div>
<div id="tmm" class="section level3">
<h3><span class="header-section-number">12.3.3</span> TMM</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi.qc &lt;-<span class="st"> </span>
<span class="st">    </span>scater::<span class="kw">normaliseExprs</span>(umi.qc,
                           <span class="dt">method =</span> <span class="st">&quot;TMM&quot;</span>,
                           <span class="dt">feature_set =</span> endog_genes,
                           <span class="dt">lib.size =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">ncol</span>(umi.qc)))
scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;norm_counts&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-tmm"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-tmm-1.png" alt="PCA plot of the blischak data after TMM normalisation" width="90%" />
<p class="caption">
Figure 12.5: PCA plot of the blischak data after TMM normalisation
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">calc_cell_RLE</span>(<span class="kw">norm_counts</span>(umi.qc)),
        <span class="dt">col =</span> <span class="st">&quot;grey50&quot;</span>,
        <span class="dt">ylab =</span> <span class="st">&quot;RLE&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;&quot;</span>, <span class="dt">ylim=</span><span class="kw">c</span>(-<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-ours-rle-tmm"></span>
<img src="11-exprs-norm_files/figure-html/norm-ours-rle-tmm-1.png" alt="Cell-wise RLE of the blischak data" width="90%" />
<p class="caption">
Figure 12.6: Cell-wise RLE of the blischak data
</p>
</div>
</div>
<div id="lsf" class="section level3">
<h3><span class="header-section-number">12.3.4</span> LSF</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qclust &lt;-<span class="st"> </span>scran::<span class="kw">quickCluster</span>(umi.qc, <span class="dt">min.size =</span> <span class="dv">30</span>)
umi.qc &lt;-<span class="st"> </span>scran::<span class="kw">computeSumFactors</span>(umi.qc, <span class="dt">sizes =</span> <span class="dv">15</span>, <span class="dt">clusters =</span> qclust)
umi.qc &lt;-<span class="st"> </span>scater::<span class="kw">normalize</span>(umi.qc)
scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;exprs&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-lsf"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-lsf-1.png" alt="PCA plot of the blischak data after LSF normalisation" width="90%" />
<p class="caption">
Figure 12.7: PCA plot of the blischak data after LSF normalisation
</p>
</div>
<p><strong>Exercise:</strong></p>
<p>Use <code>method = &quot;RLE&quot;</code> and <code>method = &quot;upperquartile&quot;</code> to perform size-factor and UQ normalizations and compare to the results above.</p>
</div>
<div id="size-factor-rle" class="section level3">
<h3><span class="header-section-number">12.3.5</span> Size-factor (RLE)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi.qc &lt;-<span class="st"> </span>
<span class="st">    </span>scater::<span class="kw">normaliseExprs</span>(umi.qc,
                           <span class="dt">method =</span> <span class="st">&quot;RLE&quot;</span>, 
                           <span class="dt">feature_set =</span> endog_genes,
                           <span class="dt">lib.size =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">ncol</span>(umi.qc)))
scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;norm_counts&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-rle"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-rle-1.png" alt="PCA plot of the blischak data after RLE normalisation" width="90%" />
<p class="caption">
Figure 12.8: PCA plot of the blischak data after RLE normalisation
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">calc_cell_RLE</span>(<span class="kw">norm_counts</span>(umi.qc)),
        <span class="dt">col =</span> <span class="st">&quot;grey50&quot;</span>,
        <span class="dt">ylab =</span> <span class="st">&quot;RLE&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;&quot;</span>, <span class="dt">ylim=</span><span class="kw">c</span>(-<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-ours-rle-rle"></span>
<img src="11-exprs-norm_files/figure-html/norm-ours-rle-rle-1.png" alt="Cell-wise RLE of the blischak data" width="90%" />
<p class="caption">
Figure 12.9: Cell-wise RLE of the blischak data
</p>
</div>
</div>
<div id="upperquantile" class="section level3">
<h3><span class="header-section-number">12.3.6</span> Upperquantile</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi.qc &lt;-<span class="st"> </span>
<span class="st">    </span>scater::<span class="kw">normaliseExprs</span>(umi.qc,
                           <span class="dt">method =</span> <span class="st">&quot;upperquartile&quot;</span>, 
                           <span class="dt">feature_set =</span> endog_genes,
                           <span class="dt">p =</span> <span class="fl">0.99</span>,
                           <span class="dt">lib.size =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">ncol</span>(umi.qc)))
scater::<span class="kw">plotPCA</span>(umi.qc[endog_genes, ],
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;norm_counts&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-uq"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-uq-1.png" alt="PCA plot of the blischak data after UQ normalisation" width="90%" />
<p class="caption">
Figure 12.10: PCA plot of the blischak data after UQ normalisation
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">calc_cell_RLE</span>(<span class="kw">norm_counts</span>(umi.qc)),
        <span class="dt">col =</span> <span class="st">&quot;grey50&quot;</span>,
        <span class="dt">ylab =</span> <span class="st">&quot;RLE&quot;</span>,
        <span class="dt">main =</span> <span class="st">&quot;&quot;</span>, <span class="dt">ylim=</span><span class="kw">c</span>(-<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-ours-rle-uq"></span>
<img src="11-exprs-norm_files/figure-html/norm-ours-rle-uq-1.png" alt="Cell-wise RLE of the blischak data" width="90%" />
<p class="caption">
Figure 12.11: Cell-wise RLE of the blischak data
</p>
</div>
</div>
</div>
<div id="other-methods" class="section level2">
<h2><span class="header-section-number">12.4</span> Other methods</h2>
<p>Some methods combine library size and fragment/gene length normalization such as:</p>
<ul>
<li><strong>RPKM</strong> - Reads Per Kilobase Million (for single-end sequencing)</li>
<li><strong>FPKM</strong> - Fragments Per Kilobase Million (same as <strong>RPKM</strong> but for paired-end sequencing, makes sure that paired ends mapped to the same fragment are not counted twice)</li>
<li><strong>TPM</strong> - Transcripts Per Kilobase Million (same as <strong>RPKM</strong>, but the order of normalizations is reversed - length first and sequencing depth second)</li>
</ul>
<p>These methods are not applicable to our dataset since the end of the transcript which contains the UMI was preferentially sequenced. Furthermore in general these should only be calculated using appropriate quantification software from aligned BAM files not from read counts since often only a portion of the entire gene/transcript is sequenced, not the entire length.</p>
<p>However, here we show how these normalisations can be calculated using scater. First, we need to find the effective transcript length in Kilobases. However, our dataset containes only gene IDs, therefore we will be using the gene lengths instead of transcripts. scater uses the <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">biomaRt</a> package, which allows one to annotate genes by other attributes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi.qc &lt;-
<span class="st">    </span>scater::<span class="kw">getBMFeatureAnnos</span>(umi.qc,
                              <span class="dt">filters =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>, 
                              <span class="dt">attributes =</span> <span class="kw">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,
                                             <span class="st">&quot;hgnc_symbol&quot;</span>,
                                             <span class="st">&quot;chromosome_name&quot;</span>,
                                             <span class="st">&quot;start_position&quot;</span>,
                                             <span class="st">&quot;end_position&quot;</span>), 
                              <span class="dt">feature_symbol =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,
                              <span class="dt">feature_id =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>,
                              <span class="dt">biomart =</span> <span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>, 
                              <span class="dt">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>,
                              <span class="dt">host =</span> <span class="st">&quot;www.ensembl.org&quot;</span>)

<span class="co"># If you have mouse data, change the arguments based on this example:</span>
<span class="co"># scater::getBMFeatureAnnos(object,</span>
<span class="co">#                           filters = &quot;ensembl_transcript_id&quot;, </span>
<span class="co">#                           attributes = c(&quot;ensembl_transcript_id&quot;, </span>
<span class="co">#                                        &quot;ensembl_gene_id&quot;, &quot;mgi_symbol&quot;, </span>
<span class="co">#                                        &quot;chromosome_name&quot;,</span>
<span class="co">#                                        &quot;transcript_biotype&quot;,</span>
<span class="co">#                                        &quot;transcript_start&quot;,</span>
<span class="co">#                                        &quot;transcript_end&quot;, </span>
<span class="co">#                                        &quot;transcript_count&quot;), </span>
<span class="co">#                           feature_symbol = &quot;mgi_symbol&quot;,</span>
<span class="co">#                           feature_id = &quot;ensembl_gene_id&quot;,</span>
<span class="co">#                           biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;, </span>
<span class="co">#                           dataset = &quot;mmusculus_gene_ensembl&quot;,</span>
<span class="co">#                           host = &quot;www.ensembl.org&quot;) </span></code></pre></div>
<p>Some of the genes were not annotated, therefore we filter them out:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">umi.qc.ann &lt;-
<span class="st">    </span>umi.qc[!<span class="kw">is.na</span>(<span class="kw">fData</span>(umi.qc)$ensembl_gene_id), ]</code></pre></div>
<p>Now we compute the total gene length in Kilobases by using the <code>end_position</code> and <code>start_position</code> fields:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eff_length &lt;-<span class="st"> </span><span class="kw">abs</span>(<span class="kw">fData</span>(umi.qc.ann)$end_position -
<span class="st">                      </span><span class="kw">fData</span>(umi.qc.ann)$start_position)/<span class="dv">1000</span></code></pre></div>
<p>Now we are ready to perform the normalisations:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">tpm</span>(umi.qc.ann) &lt;-
<span class="st">    </span><span class="kw">calculateTPM</span>(
        umi.qc.ann,
        eff_length
    )
<span class="kw">fpkm</span>(umi.qc.ann) &lt;-
<span class="st">    </span><span class="kw">calculateFPKM</span>(
        umi.qc.ann,
        eff_length
    )</code></pre></div>
<p>Plot the results as a PCA plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPCA</span>(umi.qc.ann,
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;fpkm&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-fpkm"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-fpkm-1.png" alt="PCA plot of the blischak data after FPKM normalisation" width="90%" />
<p class="caption">
Figure 12.12: PCA plot of the blischak data after FPKM normalisation
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">scater::<span class="kw">plotPCA</span>(umi.qc.ann,
                <span class="dt">colour_by =</span> <span class="st">&quot;batch&quot;</span>,
                <span class="dt">size_by =</span> <span class="st">&quot;total_features&quot;</span>,
                <span class="dt">shape_by =</span> <span class="st">&quot;individual&quot;</span>,
                <span class="dt">exprs_values =</span> <span class="st">&quot;tpm&quot;</span>)</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:norm-pca-tpm"></span>
<img src="11-exprs-norm_files/figure-html/norm-pca-tpm-1.png" alt="PCA plot of the blischak data after TPM normalisation" width="90%" />
<p class="caption">
Figure 12.13: PCA plot of the blischak data after TPM normalisation
</p>
</div>
<p><strong>Note:</strong> The PCA primarily looks for difference between cells. Since gene length is the same across cells for each gene FPKM is almost identical to the CPM plot (it is just rotated). However, if we plot the mean expression vs gene length it is apparent that normalizing for gene length should not be performed for this dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(eff_length, <span class="kw">rowMeans</span>(<span class="kw">counts</span>(umi.qc.ann)))</code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:length-vs-mean"></span>
<img src="11-exprs-norm_files/figure-html/length-vs-mean-1.png" alt="Gene length vs Mean Expression for the raw data" width="90%" />
<p class="caption">
Figure 12.14: Gene length vs Mean Expression for the raw data
</p>
</div>
</div>
<div id="exercise-1" class="section level2">
<h2><span class="header-section-number">12.5</span> Exercise</h2>
<p>Perform the same analysis with read counts of the Blischak data. Use <code>blischak/reads.rds</code> file to load the reads SCESet object. Once you have finished please compare your results to ours (next chapter).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="confounding-factors-reads.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="normalization-for-library-size-reads.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/11-exprs-norm.Rmd",
"text": "Edit"
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
