<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Analysis of single-cell RNA-seq data</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="Analysis of single-cell RNA-seq data">
  <meta name="generator" content="bookdown 0.0.73 and GitBook 2.6.7">

  <meta property="og:title" content="Analysis of single-cell RNA-seq data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Analysis of single-cell RNA-seq data" />
  
  
  

<meta name="author" content="Vladimir Kiselev, Tallulah Andrews, Davis McCarthy and Martin Hemberg">

<meta name="date" content="2016-06-18">

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="identification-of-important-genes.html">
<link rel="next" href="differential-expression-de-analysis.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Google Analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-71525309-1', 'auto');
  ga('send', 'pageview');

</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="index.html">Table of Contents</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About the course</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#registration"><i class="fa fa-check"></i><b>1.1</b> Registration</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#github"><i class="fa fa-check"></i><b>1.2</b> GitHub</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#prerequisites"><i class="fa fa-check"></i><b>1.4</b> Prerequisites</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i><b>1.5</b> Contact</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="technical-requirements.html"><a href="technical-requirements.html"><i class="fa fa-check"></i><b>2</b> Technical requirements</a><ul>
<li class="chapter" data-level="2.1" data-path="technical-requirements.html"><a href="technical-requirements.html#general"><i class="fa fa-check"></i><b>2.1</b> General</a></li>
<li class="chapter" data-level="2.2" data-path="technical-requirements.html"><a href="technical-requirements.html#plotting"><i class="fa fa-check"></i><b>2.2</b> Plotting</a></li>
<li class="chapter" data-level="2.3" data-path="technical-requirements.html"><a href="technical-requirements.html#qc-and-normalisation"><i class="fa fa-check"></i><b>2.3</b> QC and normalisation</a></li>
<li class="chapter" data-level="2.4" data-path="technical-requirements.html"><a href="technical-requirements.html#clustering"><i class="fa fa-check"></i><b>2.4</b> Clustering</a></li>
<li class="chapter" data-level="2.5" data-path="technical-requirements.html"><a href="technical-requirements.html#dropouts"><i class="fa fa-check"></i><b>2.5</b> Dropouts</a></li>
<li class="chapter" data-level="2.6" data-path="technical-requirements.html"><a href="technical-requirements.html#pseudotime"><i class="fa fa-check"></i><b>2.6</b> Pseudotime</a></li>
<li class="chapter" data-level="2.7" data-path="technical-requirements.html"><a href="technical-requirements.html#differential-expression"><i class="fa fa-check"></i><b>2.7</b> Differential Expression</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html"><i class="fa fa-check"></i><b>3</b> Introduction to single-cell RNA-seq</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#bulk-rna-seq"><i class="fa fa-check"></i><b>3.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#scrna-seq"><i class="fa fa-check"></i><b>3.2</b> scRNA-seq</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#protocol"><i class="fa fa-check"></i><b>3.3</b> Protocol</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#computational-analysis"><i class="fa fa-check"></i><b>3.4</b> Computational Analysis</a></li>
<li class="chapter" data-level="3.5" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#challenges"><i class="fa fa-check"></i><b>3.5</b> Challenges</a></li>
<li class="chapter" data-level="3.6" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#controls"><i class="fa fa-check"></i><b>3.6</b> Controls</a><ul>
<li class="chapter" data-level="3.6.1" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#spike-ins"><i class="fa fa-check"></i><b>3.6.1</b> Spike-ins</a></li>
<li class="chapter" data-level="3.6.2" data-path="introduction-to-single-cell-rna-seq.html"><a href="introduction-to-single-cell-rna-seq.html#umis"><i class="fa fa-check"></i><b>3.6.2</b> UMIs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html"><i class="fa fa-check"></i><b>4</b> Construction of expression matrix</a><ul>
<li class="chapter" data-level="4.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-qc"><i class="fa fa-check"></i><b>4.1</b> Reads QC</a></li>
<li class="chapter" data-level="4.2" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-alignment"><i class="fa fa-check"></i><b>4.2</b> Reads alignment</a></li>
<li class="chapter" data-level="4.3" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#alignment-example"><i class="fa fa-check"></i><b>4.3</b> Alignment example</a></li>
<li class="chapter" data-level="4.4" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#mapping-qc"><i class="fa fa-check"></i><b>4.4</b> Mapping QC</a></li>
<li class="chapter" data-level="4.5" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#reads-quantification"><i class="fa fa-check"></i><b>4.5</b> Reads quantification</a><ul>
<li class="chapter" data-level="4.5.1" data-path="construction-of-expression-matrix.html"><a href="construction-of-expression-matrix.html#umi-quantification"><i class="fa fa-check"></i><b>4.5.1</b> UMI quantification</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="scater-package.html"><a href="scater-package.html"><i class="fa fa-check"></i><b>5</b> scater package</a><ul>
<li class="chapter" data-level="5.1" data-path="scater-package.html"><a href="scater-package.html#introduction"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="scater-package.html"><a href="scater-package.html#scater-workflow"><i class="fa fa-check"></i><b>5.2</b> scater workflow</a></li>
<li class="chapter" data-level="5.3" data-path="scater-package.html"><a href="scater-package.html#terminology"><i class="fa fa-check"></i><b>5.3</b> Terminology</a></li>
<li class="chapter" data-level="5.4" data-path="scater-package.html"><a href="scater-package.html#sceset-class"><i class="fa fa-check"></i><b>5.4</b> SCESet class</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html"><i class="fa fa-check"></i><b>6</b> Expression QC (UMI)</a><ul>
<li class="chapter" data-level="6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#introduction-1"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#blischak-dataset"><i class="fa fa-check"></i><b>6.2</b> Blischak dataset</a></li>
<li class="chapter" data-level="6.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-qc"><i class="fa fa-check"></i><b>6.3</b> Cell QC</a><ul>
<li class="chapter" data-level="6.3.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#library-size"><i class="fa fa-check"></i><b>6.3.1</b> Library size</a></li>
<li class="chapter" data-level="6.3.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#detected-genes-1"><i class="fa fa-check"></i><b>6.3.2</b> Detected genes (1)</a></li>
<li class="chapter" data-level="6.3.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#erccs-and-mts"><i class="fa fa-check"></i><b>6.3.3</b> ERCCs and MTs</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#cell-filtering"><i class="fa fa-check"></i><b>6.4</b> Cell filtering</a><ul>
<li class="chapter" data-level="6.4.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#default-thresholds"><i class="fa fa-check"></i><b>6.4.1</b> Default Thresholds</a></li>
<li class="chapter" data-level="6.4.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#automatic"><i class="fa fa-check"></i><b>6.4.2</b> Automatic</a></li>
<li class="chapter" data-level="6.4.3" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#manual"><i class="fa fa-check"></i><b>6.4.3</b> Manual</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#compare-filterings"><i class="fa fa-check"></i><b>6.5</b> Compare filterings</a></li>
<li class="chapter" data-level="6.6" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-analysis"><i class="fa fa-check"></i><b>6.6</b> Gene analysis</a><ul>
<li class="chapter" data-level="6.6.1" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-expression"><i class="fa fa-check"></i><b>6.6.1</b> Gene expression</a></li>
<li class="chapter" data-level="6.6.2" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#gene-filtering"><i class="fa fa-check"></i><b>6.6.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#save-the-data"><i class="fa fa-check"></i><b>6.7</b> Save the data</a></li>
<li class="chapter" data-level="6.8" data-path="expression-qc-umi.html"><a href="expression-qc-umi.html#big-exercise"><i class="fa fa-check"></i><b>6.8</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="expression-qc-reads.html"><a href="expression-qc-reads.html"><i class="fa fa-check"></i><b>7</b> Expression QC (Reads)</a></li>
<li class="chapter" data-level="8" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>8</b> Data visualization</a><ul>
<li class="chapter" data-level="8.1" data-path="data-visualization.html"><a href="data-visualization.html#introduction-2"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="data-visualization.html"><a href="data-visualization.html#visual-pca"><i class="fa fa-check"></i><b>8.2</b> PCA plot</a><ul>
<li class="chapter" data-level="8.2.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc"><i class="fa fa-check"></i><b>8.2.1</b> Before QC</a></li>
<li class="chapter" data-level="8.2.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc"><i class="fa fa-check"></i><b>8.2.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="data-visualization.html"><a href="data-visualization.html#visual-tsne"><i class="fa fa-check"></i><b>8.3</b> tSNE map</a><ul>
<li class="chapter" data-level="8.3.1" data-path="data-visualization.html"><a href="data-visualization.html#before-qc-1"><i class="fa fa-check"></i><b>8.3.1</b> Before QC</a></li>
<li class="chapter" data-level="8.3.2" data-path="data-visualization.html"><a href="data-visualization.html#after-qc-1"><i class="fa fa-check"></i><b>8.3.2</b> After QC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="data-visualization.html"><a href="data-visualization.html#big-exercise-1"><i class="fa fa-check"></i><b>8.4</b> Big Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="expression-overview-reads.html"><a href="expression-overview-reads.html"><i class="fa fa-check"></i><b>9</b> Expression overview (Reads)</a></li>
<li class="chapter" data-level="10" data-path="confounding-factors.html"><a href="confounding-factors.html"><i class="fa fa-check"></i><b>10</b> Confounding factors</a><ul>
<li class="chapter" data-level="10.1" data-path="confounding-factors.html"><a href="confounding-factors.html#introduction-3"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="confounding-factors.html"><a href="confounding-factors.html#correlations-with-pcs"><i class="fa fa-check"></i><b>10.2</b> Correlations with PCs</a><ul>
<li class="chapter" data-level="10.2.1" data-path="confounding-factors.html"><a href="confounding-factors.html#detected-genes"><i class="fa fa-check"></i><b>10.2.1</b> Detected genes</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="confounding-factors.html"><a href="confounding-factors.html#explanatory-variables"><i class="fa fa-check"></i><b>10.3</b> Explanatory variables</a></li>
<li class="chapter" data-level="10.4" data-path="confounding-factors.html"><a href="confounding-factors.html#other-confounders"><i class="fa fa-check"></i><b>10.4</b> Other confounders</a></li>
<li class="chapter" data-level="10.5" data-path="confounding-factors.html"><a href="confounding-factors.html#exercise"><i class="fa fa-check"></i><b>10.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="confounding-factors-reads.html"><a href="confounding-factors-reads.html"><i class="fa fa-check"></i><b>11</b> Confounding factors (Reads)</a></li>
<li class="chapter" data-level="12" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html"><i class="fa fa-check"></i><b>12</b> Normalization for library size</a><ul>
<li class="chapter" data-level="12.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#introduction-4"><i class="fa fa-check"></i><b>12.1</b> Introduction</a></li>
<li class="chapter" data-level="12.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#library-size-1"><i class="fa fa-check"></i><b>12.2</b> Library size</a></li>
<li class="chapter" data-level="12.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#normalisations"><i class="fa fa-check"></i><b>12.3</b> Normalisations</a><ul>
<li class="chapter" data-level="12.3.1" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#raw"><i class="fa fa-check"></i><b>12.3.1</b> Raw</a></li>
<li class="chapter" data-level="12.3.2" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#cpm"><i class="fa fa-check"></i><b>12.3.2</b> CPM</a></li>
<li class="chapter" data-level="12.3.3" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#tmm"><i class="fa fa-check"></i><b>12.3.3</b> TMM</a></li>
<li class="chapter" data-level="12.3.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#scran"><i class="fa fa-check"></i><b>12.3.4</b> scran</a></li>
<li class="chapter" data-level="12.3.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#size-factor-rle"><i class="fa fa-check"></i><b>12.3.5</b> Size-factor (RLE)</a></li>
<li class="chapter" data-level="12.3.6" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#upperquantile"><i class="fa fa-check"></i><b>12.3.6</b> Upperquantile</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#other-methods"><i class="fa fa-check"></i><b>12.4</b> Other methods</a></li>
<li class="chapter" data-level="12.5" data-path="normalization-for-library-size.html"><a href="normalization-for-library-size.html#exercise-1"><i class="fa fa-check"></i><b>12.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="normalization-for-library-size-reads.html"><a href="normalization-for-library-size-reads.html"><i class="fa fa-check"></i><b>13</b> Normalization for library size (Reads)</a></li>
<li class="chapter" data-level="14" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html"><i class="fa fa-check"></i><b>14</b> Remove confounders using controls</a><ul>
<li class="chapter" data-level="14.1" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#introduction-5"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#remove-unwanted-variation"><i class="fa fa-check"></i><b>14.2</b> Remove Unwanted Variation</a></li>
<li class="chapter" data-level="14.3" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-1"><i class="fa fa-check"></i><b>14.3</b> Effectiveness 1</a></li>
<li class="chapter" data-level="14.4" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#effectiveness-2"><i class="fa fa-check"></i><b>14.4</b> Effectiveness 2</a></li>
<li class="chapter" data-level="14.5" data-path="remove-confounders-using-controls.html"><a href="remove-confounders-using-controls.html#exercise-2"><i class="fa fa-check"></i><b>14.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="remove-confounders-using-controls-reads.html"><a href="remove-confounders-using-controls-reads.html"><i class="fa fa-check"></i><b>15</b> Remove confounders using controls (Reads)</a></li>
<li class="chapter" data-level="16" data-path="clustering-analysis.html"><a href="clustering-analysis.html"><i class="fa fa-check"></i><b>16</b> Clustering analysis</a><ul>
<li class="chapter" data-level="16.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#introduction-6"><i class="fa fa-check"></i><b>16.1</b> Introduction</a></li>
<li class="chapter" data-level="16.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#dimensionality-reductions"><i class="fa fa-check"></i><b>16.2</b> Dimensionality reductions</a><ul>
<li class="chapter" data-level="16.2.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pca"><i class="fa fa-check"></i><b>16.2.1</b> PCA</a></li>
<li class="chapter" data-level="16.2.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#spectral"><i class="fa fa-check"></i><b>16.2.2</b> Spectral</a></li>
<li class="chapter" data-level="16.2.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne"><i class="fa fa-check"></i><b>16.2.3</b> tSNE</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#clustering-methods"><i class="fa fa-check"></i><b>16.3</b> Clustering methods</a><ul>
<li class="chapter" data-level="16.3.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>16.3.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="16.3.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#k-means"><i class="fa fa-check"></i><b>16.3.2</b> k-means</a></li>
<li class="chapter" data-level="16.3.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#graph-based-methods"><i class="fa fa-check"></i><b>16.3.3</b> Graph-based methods</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#challenges-in-clustering"><i class="fa fa-check"></i><b>16.4</b> Challenges in clustering</a></li>
<li class="chapter" data-level="16.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tools-for-scrna-seq-data"><i class="fa fa-check"></i><b>16.5</b> Tools for scRNA-seq data</a><ul>
<li class="chapter" data-level="16.5.1" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sincera"><i class="fa fa-check"></i><b>16.5.1</b> <a href="https://research.cchmc.org/pbge/sincera.html">SINCERA</a></a></li>
<li class="chapter" data-level="16.5.2" data-path="clustering-analysis.html"><a href="clustering-analysis.html#pcareduce"><i class="fa fa-check"></i><b>16.5.2</b> <a href="https://github.com/JustinaZ/pcaReduce">pcaReduce</a></a></li>
<li class="chapter" data-level="16.5.3" data-path="clustering-analysis.html"><a href="clustering-analysis.html#sc3"><i class="fa fa-check"></i><b>16.5.3</b> <a href="http://bioconductor.org/packages/SC3/">SC3</a></a></li>
<li class="chapter" data-level="16.5.4" data-path="clustering-analysis.html"><a href="clustering-analysis.html#tsne-k-means"><i class="fa fa-check"></i><b>16.5.4</b> tSNE + k-means</a></li>
<li class="chapter" data-level="16.5.5" data-path="clustering-analysis.html"><a href="clustering-analysis.html#seurat"><i class="fa fa-check"></i><b>16.5.5</b> <a href="https://github.com/Puriney/seurat">SEURAT</a></a></li>
<li class="chapter" data-level="16.5.6" data-path="clustering-analysis.html"><a href="clustering-analysis.html#snn-cliq"><i class="fa fa-check"></i><b>16.5.6</b> <a href="http://bioinfo.uncc.edu/SNNCliq/">SNN-Cliq</a></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="clust-methods.html"><a href="clust-methods.html"><i class="fa fa-check"></i><b>17</b> Clustering example</a><ul>
<li class="chapter" data-level="17.1" data-path="clust-methods.html"><a href="clust-methods.html#patient-dataset"><i class="fa fa-check"></i><b>17.1</b> Patient dataset</a></li>
<li class="chapter" data-level="17.2" data-path="clust-methods.html"><a href="clust-methods.html#sc3-1"><i class="fa fa-check"></i><b>17.2</b> SC3</a></li>
<li class="chapter" data-level="17.3" data-path="clust-methods.html"><a href="clust-methods.html#pcareduce-1"><i class="fa fa-check"></i><b>17.3</b> pcaReduce</a></li>
<li class="chapter" data-level="17.4" data-path="clust-methods.html"><a href="clust-methods.html#tsne-kmeans"><i class="fa fa-check"></i><b>17.4</b> tSNE + kmeans</a></li>
<li class="chapter" data-level="17.5" data-path="clust-methods.html"><a href="clust-methods.html#snn-cliq-1"><i class="fa fa-check"></i><b>17.5</b> SNN-Cliq</a></li>
<li class="chapter" data-level="17.6" data-path="clust-methods.html"><a href="clust-methods.html#sincera-1"><i class="fa fa-check"></i><b>17.6</b> SINCERA</a></li>
<li class="chapter" data-level="17.7" data-path="clust-methods.html"><a href="clust-methods.html#seurat-1"><i class="fa fa-check"></i><b>17.7</b> SEURAT</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html"><i class="fa fa-check"></i><b>18</b> Identification of important genes</a><ul>
<li class="chapter" data-level="18.1" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#fitting-the-models"><i class="fa fa-check"></i><b>18.1</b> Fitting the models</a></li>
<li class="chapter" data-level="18.2" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#the-michaelis-menten-equation"><i class="fa fa-check"></i><b>18.2</b> The Michaelis-Menten Equation</a></li>
<li class="chapter" data-level="18.3" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#right-outliers"><i class="fa fa-check"></i><b>18.3</b> Right outliers</a></li>
<li class="chapter" data-level="18.4" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#validation-of-de-results"><i class="fa fa-check"></i><b>18.4</b> Validation of DE results</a></li>
<li class="chapter" data-level="18.5" data-path="identification-of-important-genes.html"><a href="identification-of-important-genes.html#comparing-m3drop-to-other-methods"><i class="fa fa-check"></i><b>18.5</b> Comparing M3Drop to other methods</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="ordering-of-cells-according-to-pseudotime.html"><a href="ordering-of-cells-according-to-pseudotime.html"><i class="fa fa-check"></i><b>19</b> Ordering of cells according to pseudotime</a><ul>
<li class="chapter" data-level="19.1" data-path="ordering-of-cells-according-to-pseudotime.html"><a href="ordering-of-cells-according-to-pseudotime.html#tscan"><i class="fa fa-check"></i><b>19.1</b> TSCAN</a></li>
<li class="chapter" data-level="19.2" data-path="ordering-of-cells-according-to-pseudotime.html"><a href="ordering-of-cells-according-to-pseudotime.html#monocle"><i class="fa fa-check"></i><b>19.2</b> monocle</a></li>
<li class="chapter" data-level="19.3" data-path="ordering-of-cells-according-to-pseudotime.html"><a href="ordering-of-cells-according-to-pseudotime.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>19.3</b> Comparison of the methods</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html"><i class="fa fa-check"></i><b>20</b> Differential Expression (DE) analysis</a><ul>
<li class="chapter" data-level="20.1" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#bulk-rna-seq-1"><i class="fa fa-check"></i><b>20.1</b> Bulk RNA-seq</a></li>
<li class="chapter" data-level="20.2" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-cell-rna-seq"><i class="fa fa-check"></i><b>20.2</b> Single cell RNA-seq</a></li>
<li class="chapter" data-level="20.3" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#scrna-seq-synthetic-data"><i class="fa fa-check"></i><b>20.3</b> scRNA-seq synthetic data</a></li>
<li class="chapter" data-level="20.4" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#single-gene-expression"><i class="fa fa-check"></i><b>20.4</b> Single gene expression</a></li>
<li class="chapter" data-level="20.5" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#poisson-beta-distribution"><i class="fa fa-check"></i><b>20.5</b> Poisson-Beta distribution</a></li>
<li class="chapter" data-level="20.6" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#sample-size"><i class="fa fa-check"></i><b>20.6</b> Sample size</a></li>
<li class="chapter" data-level="20.7" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#dropout-noise"><i class="fa fa-check"></i><b>20.7</b> Dropout noise</a></li>
<li class="chapter" data-level="20.8" data-path="differential-expression-de-analysis.html"><a href="differential-expression-de-analysis.html#dispersion-noise"><i class="fa fa-check"></i><b>20.8</b> Dispersion noise</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html"><i class="fa fa-check"></i><b>21</b> DE in a synthetic dataset</a><ul>
<li class="chapter" data-level="21.1" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#dataset"><i class="fa fa-check"></i><b>21.1</b> Dataset</a></li>
<li class="chapter" data-level="21.2" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#de-in-scrna-seq"><i class="fa fa-check"></i><b>21.2</b> DE in scRNA-seq</a></li>
<li class="chapter" data-level="21.3" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#kolmogorov-smirnov-test"><i class="fa fa-check"></i><b>21.3</b> Kolmogorov-Smirnov test</a></li>
<li class="chapter" data-level="21.4" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#performance-of-ks-test"><i class="fa fa-check"></i><b>21.4</b> Performance of KS test</a></li>
<li class="chapter" data-level="21.5" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#deseq2"><i class="fa fa-check"></i><b>21.5</b> DESeq2</a></li>
<li class="chapter" data-level="21.6" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#scde"><i class="fa fa-check"></i><b>21.6</b> SCDE</a></li>
<li class="chapter" data-level="21.7" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#comparison-of-the-methods-1"><i class="fa fa-check"></i><b>21.7</b> Comparison of the methods</a></li>
<li class="chapter" data-level="21.8" data-path="de-in-a-synthetic-dataset.html"><a href="de-in-a-synthetic-dataset.html#beyond-changes-in-the-mean"><i class="fa fa-check"></i><b>21.8</b> Beyond changes in the mean</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html"><i class="fa fa-check"></i><b>22</b> DE in a real dataset</a><ul>
<li class="chapter" data-level="22.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#introduction-7"><i class="fa fa-check"></i><b>22.1</b> Introduction</a></li>
<li class="chapter" data-level="22.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#ks-test"><i class="fa fa-check"></i><b>22.2</b> KS-test</a></li>
<li class="chapter" data-level="22.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#deseq2-1"><i class="fa fa-check"></i><b>22.3</b> DESeq2</a></li>
<li class="chapter" data-level="22.4" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#scde-1"><i class="fa fa-check"></i><b>22.4</b> SCDE</a></li>
<li class="chapter" data-level="22.5" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#comparison-of-the-methods-2"><i class="fa fa-check"></i><b>22.5</b> Comparison of the methods</a></li>
<li class="chapter" data-level="22.6" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#visualisation"><i class="fa fa-check"></i><b>22.6</b> Visualisation</a><ul>
<li class="chapter" data-level="22.6.1" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#volcano-plots"><i class="fa fa-check"></i><b>22.6.1</b> Volcano plots</a></li>
<li class="chapter" data-level="22.6.2" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#ma-plot"><i class="fa fa-check"></i><b>22.6.2</b> MA-plot</a></li>
<li class="chapter" data-level="22.6.3" data-path="de-in-a-real-dataset.html"><a href="de-in-a-real-dataset.html#heatmap-of-de-genes"><i class="fa fa-check"></i><b>22.6.3</b> Heatmap of DE genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="advanced-exercises.html"><a href="advanced-exercises.html"><i class="fa fa-check"></i><b>23</b> Advanced exercises</a></li>
<li class="chapter" data-level="24" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>24</b> Resources</a><ul>
<li class="chapter" data-level="24.1" data-path="resources.html"><a href="resources.html#scrna-seq-protocols"><i class="fa fa-check"></i><b>24.1</b> scRNA-seq protocols</a></li>
<li class="chapter" data-level="24.2" data-path="resources.html"><a href="resources.html#scrna-seq-analysis-tools"><i class="fa fa-check"></i><b>24.2</b> scRNA-seq analysis tools</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>25</b> References</a></li>
<li class="divider"></li>
<li><a href="http://www.sanger.ac.uk/science/groups/hemberg-group" target="blank">Hemberg Lab, 2016</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Analysis of single-cell RNA-seq data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ordering-of-cells-according-to-pseudotime" class="section level1">
<h1><span class="header-section-number">19</span> Ordering of cells according to pseudotime</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TSCAN)
<span class="kw">library</span>(M3Drop)
<span class="kw">library</span>(monocle)
<span class="kw">set.seed</span>(<span class="dv">1</span>)</code></pre></div>
<p>In many situations, one is studying a process where cells change continuously. This includes for example many differentiation processes taking place during development, where following a stimulus, cells will change from one cell-type to another. Ideally, we would like to monitor the expression levels of an individual cell over time. Unfortunately, such monitoring is not possible with scRNA-seq since the cell is lysed (destroyed) when the RNA is extracted.</p>
<p>Instead, we must sample at multiple time-points and obtain snapshots of the gene expression profiles. Since some of the cells will proceed faster along the differentiation than others, each snapshot may contain cells at varying points along the developmental progression. We use statistical methods to order the cells along one or more trajectories which represent the underlying developmental trajectories, this ordering is referred to a “pseudotime”.</p>
<p>In this chapter we will consider two different tools, Monocle and TSCAN for ordering cells according to their pseudotime development. To illustrate the methods we will be using a dataset on mouse embryonic development that was collected by Deng et al. The dataset consists of 268 cells from 10 different time-points of early mouse development.</p>
<div id="tscan" class="section level2">
<h2><span class="header-section-number">19.1</span> TSCAN</h2>
<p>TSCAN combines clustering with pseudotime analysis. First it clustering the cells using <code>mclust</code>, which is based on a mixture of normal distributions. Then it builds a minimum spanning tree to connect the clusters together. The branch of this tree which connects the largest number of clusters is the main branch which is used to determine pseudotime.</p>
<p>First we will try to use all genes to order the cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">deng &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;deng/deng.rds&quot;</span>)
cellLabels &lt;-<span class="st"> </span><span class="kw">colnames</span>(deng)
procdeng &lt;-<span class="st"> </span>TSCAN::<span class="kw">preprocess</span>(deng, <span class="dt">clusternum=</span><span class="dv">10</span>)
<span class="kw">colnames</span>(procdeng) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="dv">268</span>
dengclust &lt;-<span class="st"> </span>TSCAN::<span class="kw">exprmclust</span>(procdeng, <span class="dt">clusternum=</span><span class="dv">10</span>)
TSCAN::<span class="kw">plotmclust</span>(dengclust)</code></pre></div>
<p><img src="18-pseudotime_files/figure-html/tscan-all-genes-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dengorderTSCAN &lt;-<span class="st"> </span>TSCAN::<span class="kw">TSCANorder</span>(dengclust, <span class="dt">orderonly=</span>F)
pseudotime_order_tscan &lt;-<span class="st"> </span><span class="kw">as.character</span>(dengorderTSCAN$sample_name)
pseudotime_order_tscan</code></pre></div>
<pre><code>##   [1] &quot;9&quot;   &quot;12&quot;  &quot;3&quot;   &quot;2&quot;   &quot;8&quot;   &quot;4&quot;   &quot;1&quot;   &quot;6&quot;   &quot;10&quot;  &quot;11&quot;  &quot;7&quot;  
##  [12] &quot;5&quot;   &quot;16&quot;  &quot;15&quot;  &quot;18&quot;  &quot;14&quot;  &quot;17&quot;  &quot;19&quot;  &quot;23&quot;  &quot;24&quot;  &quot;21&quot;  &quot;20&quot; 
##  [23] &quot;22&quot;  &quot;29&quot;  &quot;28&quot;  &quot;30&quot;  &quot;32&quot;  &quot;13&quot;  &quot;31&quot;  &quot;27&quot;  &quot;34&quot;  &quot;25&quot;  &quot;33&quot; 
##  [34] &quot;26&quot;  &quot;40&quot;  &quot;47&quot;  &quot;45&quot;  &quot;46&quot;  &quot;48&quot;  &quot;37&quot;  &quot;43&quot;  &quot;38&quot;  &quot;44&quot;  &quot;39&quot; 
##  [45] &quot;41&quot;  &quot;81&quot;  &quot;85&quot;  &quot;111&quot; &quot;97&quot;  &quot;104&quot; &quot;96&quot;  &quot;128&quot; &quot;87&quot;  &quot;93&quot;  &quot;71&quot; 
##  [56] &quot;99&quot;  &quot;114&quot; &quot;116&quot; &quot;60&quot;  &quot;64&quot;  &quot;105&quot; &quot;117&quot; &quot;100&quot; &quot;119&quot; &quot;51&quot;  &quot;115&quot;
##  [67] &quot;108&quot; &quot;50&quot;  &quot;106&quot; &quot;89&quot;  &quot;132&quot; &quot;112&quot; &quot;58&quot;  &quot;75&quot;  &quot;63&quot;  &quot;66&quot;  &quot;65&quot; 
##  [78] &quot;61&quot;  &quot;133&quot; &quot;120&quot; &quot;59&quot;  &quot;94&quot;  &quot;68&quot;  &quot;52&quot;  &quot;80&quot;  &quot;82&quot;  &quot;79&quot;  &quot;67&quot; 
##  [89] &quot;91&quot;  &quot;88&quot;  &quot;124&quot; &quot;56&quot;  &quot;122&quot; &quot;62&quot;  &quot;121&quot; &quot;126&quot; &quot;125&quot; &quot;69&quot;  &quot;113&quot;
## [100] &quot;123&quot; &quot;84&quot;  &quot;36&quot;  &quot;35&quot;  &quot;42&quot;  &quot;57&quot;  &quot;90&quot;  &quot;86&quot;  &quot;107&quot; &quot;92&quot;  &quot;70&quot; 
## [111] &quot;135&quot; &quot;98&quot;  &quot;78&quot;  &quot;73&quot;  &quot;76&quot;  &quot;53&quot;  &quot;95&quot;  &quot;74&quot;  &quot;129&quot; &quot;55&quot;  &quot;49&quot; 
## [122] &quot;130&quot; &quot;77&quot;  &quot;127&quot; &quot;54&quot;  &quot;134&quot; &quot;83&quot;  &quot;208&quot; &quot;228&quot; &quot;173&quot; &quot;205&quot; &quot;153&quot;
## [133] &quot;142&quot; &quot;72&quot;  &quot;164&quot; &quot;207&quot; &quot;152&quot; &quot;179&quot; &quot;227&quot; &quot;165&quot; &quot;144&quot; &quot;169&quot; &quot;219&quot;
## [144] &quot;171&quot; &quot;262&quot; &quot;170&quot; &quot;199&quot; &quot;253&quot; &quot;198&quot; &quot;200&quot; &quot;155&quot; &quot;212&quot; &quot;158&quot; &quot;197&quot;
## [155] &quot;258&quot; &quot;157&quot; &quot;209&quot; &quot;156&quot; &quot;194&quot; &quot;188&quot; &quot;189&quot; &quot;252&quot; &quot;160&quot; &quot;191&quot; &quot;265&quot;
## [166] &quot;187&quot; &quot;190&quot; &quot;195&quot; &quot;220&quot; &quot;263&quot; &quot;161&quot; &quot;213&quot; &quot;214&quot; &quot;222&quot; &quot;148&quot; &quot;196&quot;
## [177] &quot;225&quot; &quot;259&quot; &quot;162&quot; &quot;183&quot; &quot;251&quot; &quot;193&quot; &quot;143&quot; &quot;182&quot; &quot;140&quot; &quot;206&quot; &quot;141&quot;
## [188] &quot;201&quot; &quot;249&quot; &quot;261&quot; &quot;181&quot; &quot;239&quot; &quot;223&quot; &quot;255&quot; &quot;159&quot; &quot;260&quot; &quot;216&quot; &quot;151&quot;
## [199] &quot;145&quot; &quot;235&quot; &quot;215&quot; &quot;211&quot; &quot;240&quot; &quot;268&quot; &quot;267&quot; &quot;257&quot; &quot;245&quot; &quot;256&quot; &quot;250&quot;
## [210] &quot;217&quot; &quot;244&quot; &quot;237&quot; &quot;254&quot; &quot;233&quot; &quot;186&quot; &quot;234&quot; &quot;236&quot; &quot;246&quot; &quot;203&quot; &quot;180&quot;
## [221] &quot;177&quot; &quot;149&quot; &quot;136&quot; &quot;185&quot; &quot;184&quot; &quot;231&quot; &quot;264&quot; &quot;178&quot; &quot;243&quot; &quot;204&quot; &quot;230&quot;
## [232] &quot;176&quot; &quot;147&quot; &quot;202&quot; &quot;229&quot; &quot;154&quot; &quot;247&quot; &quot;172&quot; &quot;146&quot; &quot;232&quot; &quot;167&quot; &quot;238&quot;
## [243] &quot;218&quot; &quot;266&quot; &quot;174&quot; &quot;168&quot; &quot;248&quot; &quot;241&quot; &quot;210&quot; &quot;242&quot; &quot;163&quot; &quot;224&quot; &quot;226&quot;
## [254] &quot;221&quot; &quot;139&quot; &quot;150&quot; &quot;166&quot;</code></pre>
<p>We can also examine which cells have been assigned to each state:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cellLabels[dengclust$clusterid ==<span class="st"> </span><span class="dv">1</span>]</code></pre></div>
<pre><code>##  [1] &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot; &quot;2&quot;</code></pre>
<p><strong>Exercise</strong> Compare results for different numbers of clusters (<code>clusternum</code>).</p>
</div>
<div id="monocle" class="section level2">
<h2><span class="header-section-number">19.2</span> monocle</h2>
<p>Monocle skips the clustering stage of TSCAN and directly builds a minimum spanning tree to connect all cells. Monocle then identifies the longest path in this tree to determine pseudotime. If the data contains diverging trajectories (i.e. one cell type differentiation into two different cell-types), monocle can identify alternative long paths in the tree using the argument <code>num_paths</code>. Each of the resulting forked paths is defined as a separate cell state, thus <code>num_paths=2</code> will identify three different cell states.</p>
<p>Unfortunately, Monocle does not work when all the genes are used, so we must carry out feature selection. First, we use M3Drop:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m3dGenes &lt;-<span class="st"> </span><span class="kw">as.character</span>(
    M3Drop::<span class="kw">M3Drop_Differential_Expression</span>(deng, <span class="dt">suppress.plot=</span>T)$Gene
)
d &lt;-<span class="st"> </span>deng[<span class="kw">which</span>(m3dGenes %in%<span class="st"> </span><span class="kw">rownames</span>(deng)),]
d &lt;-<span class="st"> </span>d[!<span class="kw">duplicated</span>(<span class="kw">rownames</span>(d)),]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pd &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colnames</span>(d))
<span class="kw">names</span>(pd) &lt;-<span class="st"> &quot;timepoint&quot;</span>
pd &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="dt">data=</span>pd)
fd &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">rownames</span>(d))
<span class="kw">names</span>(fd) &lt;-<span class="st"> &quot;gene&quot;</span>
fd &lt;-<span class="st"> </span><span class="kw">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="dt">data=</span>fd)
<span class="kw">colnames</span>(d) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">dim</span>(d)[<span class="dv">2</span>]
<span class="kw">rownames</span>(d) &lt;-<span class="st"> </span><span class="dv">1</span>:<span class="kw">dim</span>(d)[<span class="dv">1</span>]
dCellData &lt;-<span class="st"> </span>monocle::<span class="kw">newCellDataSet</span>(d, <span class="dt">phenoData=</span>pd, <span class="dt">featureData=</span>fd)
dCellData &lt;-<span class="st"> </span>monocle::<span class="kw">setOrderingFilter</span>(dCellData, <span class="dv">1</span>:<span class="kw">length</span>(m3dGenes))
dCellDataSet &lt;-<span class="st"> </span>monocle::<span class="kw">reduceDimension</span>(dCellData, <span class="dt">pseudo_expr=</span><span class="dv">1</span>)</code></pre></div>
<pre><code>## Reducing to independent components</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dCellDataSet &lt;-<span class="st"> </span>monocle::<span class="kw">orderCells</span>(dCellDataSet, <span class="dt">reverse=</span>T, <span class="dt">num_paths=</span><span class="dv">2</span>)
monocle::<span class="kw">plot_spanning_tree</span>(dCellDataSet)</code></pre></div>
<p><img src="18-pseudotime_files/figure-html/monocle-all-genes-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pseudotime_monocle &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Cell =</span> cellLabels, 
                                <span class="dt">time =</span> <span class="kw">phenoData</span>(dCellDataSet)$Pseudotime, 
                                <span class="dt">State=</span><span class="kw">phenoData</span>(dCellDataSet)$State)
pseudotime_order_monocle &lt;-<span class="st"> </span><span class="kw">rownames</span>(pseudotime_monocle[<span class="kw">order</span>(pseudotime_monocle$time), ])
pseudotime_order_monocle</code></pre></div>
<pre><code>##   [1] &quot;192&quot; &quot;263&quot; &quot;139&quot; &quot;153&quot; &quot;173&quot; &quot;137&quot; &quot;221&quot; &quot;266&quot; &quot;222&quot; &quot;259&quot; &quot;242&quot;
##  [12] &quot;246&quot; &quot;163&quot; &quot;260&quot; &quot;224&quot; &quot;166&quot; &quot;188&quot; &quot;238&quot; &quot;227&quot; &quot;248&quot; &quot;264&quot; &quot;234&quot;
##  [23] &quot;231&quot; &quot;261&quot; &quot;258&quot; &quot;243&quot; &quot;241&quot; &quot;268&quot; &quot;233&quot; &quot;247&quot; &quot;226&quot; &quot;252&quot; &quot;262&quot;
##  [34] &quot;267&quot; &quot;237&quot; &quot;150&quot; &quot;230&quot; &quot;175&quot; &quot;138&quot; &quot;256&quot; &quot;235&quot; &quot;257&quot; &quot;168&quot; &quot;236&quot;
##  [45] &quot;195&quot; &quot;250&quot; &quot;232&quot; &quot;265&quot; &quot;254&quot; &quot;255&quot; &quot;253&quot; &quot;178&quot; &quot;229&quot; &quot;223&quot; &quot;190&quot;
##  [56] &quot;196&quot; &quot;148&quot; &quot;152&quot; &quot;164&quot; &quot;147&quot; &quot;170&quot; &quot;155&quot; &quot;156&quot; &quot;140&quot; &quot;154&quot; &quot;186&quot;
##  [67] &quot;228&quot; &quot;167&quot; &quot;203&quot; &quot;245&quot; &quot;219&quot; &quot;174&quot; &quot;240&quot; &quot;244&quot; &quot;143&quot; &quot;180&quot; &quot;185&quot;
##  [78] &quot;218&quot; &quot;209&quot; &quot;191&quot; &quot;184&quot; &quot;239&quot; &quot;193&quot; &quot;199&quot; &quot;172&quot; &quot;151&quot; &quot;145&quot; &quot;179&quot;
##  [89] &quot;136&quot; &quot;189&quot; &quot;204&quot; &quot;187&quot; &quot;181&quot; &quot;197&quot; &quot;220&quot; &quot;225&quot; &quot;176&quot; &quot;157&quot; &quot;144&quot;
## [100] &quot;198&quot; &quot;182&quot; &quot;183&quot; &quot;200&quot; &quot;160&quot; &quot;146&quot; &quot;165&quot; &quot;159&quot; &quot;207&quot; &quot;161&quot; &quot;194&quot;
## [111] &quot;158&quot; &quot;142&quot; &quot;177&quot; &quot;171&quot; &quot;217&quot; &quot;249&quot; &quot;215&quot; &quot;251&quot; &quot;216&quot; &quot;211&quot; &quot;201&quot;
## [122] &quot;202&quot; &quot;162&quot; &quot;149&quot; &quot;210&quot; &quot;213&quot; &quot;141&quot; &quot;214&quot; &quot;206&quot; &quot;169&quot; &quot;212&quot; &quot;208&quot;
## [133] &quot;205&quot; &quot;131&quot; &quot;72&quot;  &quot;118&quot; &quot;101&quot; &quot;125&quot; &quot;135&quot; &quot;102&quot; &quot;110&quot; &quot;117&quot; &quot;124&quot;
## [144] &quot;133&quot; &quot;128&quot; &quot;109&quot; &quot;127&quot; &quot;120&quot; &quot;113&quot; &quot;134&quot; &quot;122&quot; &quot;129&quot; &quot;121&quot; &quot;126&quot;
## [155] &quot;119&quot; &quot;130&quot; &quot;132&quot; &quot;108&quot; &quot;105&quot; &quot;115&quot; &quot;114&quot; &quot;112&quot; &quot;100&quot; &quot;99&quot;  &quot;88&quot; 
## [166] &quot;123&quot; &quot;97&quot;  &quot;91&quot;  &quot;111&quot; &quot;66&quot;  &quot;70&quot;  &quot;96&quot;  &quot;71&quot;  &quot;94&quot;  &quot;63&quot;  &quot;98&quot; 
## [177] &quot;95&quot;  &quot;51&quot;  &quot;73&quot;  &quot;74&quot;  &quot;116&quot; &quot;92&quot;  &quot;61&quot;  &quot;55&quot;  &quot;54&quot;  &quot;56&quot;  &quot;93&quot; 
## [188] &quot;53&quot;  &quot;52&quot;  &quot;86&quot;  &quot;49&quot;  &quot;87&quot;  &quot;50&quot;  &quot;89&quot;  &quot;75&quot;  &quot;64&quot;  &quot;60&quot;  &quot;62&quot; 
## [199] &quot;76&quot;  &quot;79&quot;  &quot;67&quot;  &quot;57&quot;  &quot;85&quot;  &quot;104&quot; &quot;83&quot;  &quot;58&quot;  &quot;90&quot;  &quot;107&quot; &quot;106&quot;
## [210] &quot;68&quot;  &quot;65&quot;  &quot;59&quot;  &quot;77&quot;  &quot;84&quot;  &quot;82&quot;  &quot;69&quot;  &quot;78&quot;  &quot;80&quot;  &quot;81&quot;  &quot;38&quot; 
## [221] &quot;39&quot;  &quot;35&quot;  &quot;41&quot;  &quot;42&quot;  &quot;44&quot;  &quot;43&quot;  &quot;37&quot;  &quot;40&quot;  &quot;36&quot;  &quot;45&quot;  &quot;46&quot; 
## [232] &quot;47&quot;  &quot;48&quot;  &quot;26&quot;  &quot;30&quot;  &quot;33&quot;  &quot;34&quot;  &quot;25&quot;  &quot;27&quot;  &quot;29&quot;  &quot;28&quot;  &quot;31&quot; 
## [243] &quot;32&quot;  &quot;17&quot;  &quot;18&quot;  &quot;13&quot;  &quot;20&quot;  &quot;24&quot;  &quot;19&quot;  &quot;14&quot;  &quot;22&quot;  &quot;23&quot;  &quot;21&quot; 
## [254] &quot;15&quot;  &quot;16&quot;  &quot;5&quot;   &quot;8&quot;   &quot;2&quot;   &quot;1&quot;   &quot;103&quot; &quot;4&quot;   &quot;3&quot;   &quot;9&quot;   &quot;10&quot; 
## [265] &quot;11&quot;  &quot;12&quot;  &quot;7&quot;   &quot;6&quot;</code></pre>
</div>
<div id="comparison-of-the-methods" class="section level2">
<h2><span class="header-section-number">19.3</span> Comparison of the methods</h2>
<p>How do the trajectories inferred by TSCAN and Monocle compare?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">matched_ordering &lt;-<span class="st"> </span><span class="kw">match</span>(pseudotime_order_tscan, pseudotime_order_monocle)
<span class="kw">plot</span>(matched_ordering, <span class="dt">xlab =</span> <span class="st">&quot;Monocle Order&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;TSCAN Order&quot;</span>)</code></pre></div>
<p><img src="18-pseudotime_files/figure-html/tscan-monocle-compare-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Exercise</strong>: Repeat the exercise using a subset of the genes, e.g. the set of highly variable genes that can be obtained using M3Drop::Brennecke_getVariableGenes</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identification-of-important-genes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-de-analysis.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/hemberg-lab/scRNA.seq.course/edit/master/18-pseudotime.Rmd",
"text": "Edit"
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
